diff --git a/configure.in b/configure.in
index 396f3ac..a6f3da1 100644
--- a/configure.in
+++ b/configure.in
@@ -69,8 +69,14 @@ if test "x$ac_cv_ftw_actionretval" = xyes; then
 fi
 
 dnl SELinux checks
-AC_CHECK_LIB(selinux,is_selinux_enabled)
-AC_CHECK_HEADERS(selinux/selinux.h)
+AC_ARG_ENABLE([selinux],
+    AS_HELP_STRING([--enable-selinux],[enable SELinux support [default=auto]]),
+    [], [enable_selinux=auto])
+
+if test x$enable_selinux != xno; then
+    AC_CHECK_LIB(selinux,is_selinux_enabled)
+    AC_CHECK_HEADERS(selinux/selinux.h)
+fi
 
 dnl This test must come as early as possible after the compiler configuration
 dnl tests, because the choice of the file model can (in principle) affect
