diff --git a/Makefile b/Makefile
index eb79a28..0195ce3 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-PREFIX=/usr/local
+PREFIX=/usr
 BINDIR=/bin
 
 build:
@@ -9,5 +9,4 @@ clean:
 
 install: build
 	install -d $(DESTDIR)$(PREFIX)$(BINDIR)
-	install -m 0755 src/unionfs $(DESTDIR)$(PREFIX)$(BINDIR)
-	install -m 0644 man/unionfs-fuse.8 $(DESTDIR)$(PREFIX)/share/man/man8/
+	install -m 0755 src/unionfs $(DESTDIR)$(PREFIX)$(BINDIR)/unionfs-fuse
