diff --git a/libexslt.pc.in b/libexslt.pc.in
index 2f6378f..3f6153b 100644
--- a/libexslt.pc.in
+++ b/libexslt.pc.in
@@ -8,6 +8,6 @@ Name: libexslt
 Version: @LIBEXSLT_VERSION@
 Description: EXSLT Extension library
 Requires: libxml-2.0
-Libs: @EXSLT_LIBDIR@ @EXSLT_LIBS@
+Libs: -lexslt @EXSLT_LIBDIR@ @EXSLT_LIBS@
 Libs.private: @M_LIBS@ @LIBGCRYPT_LIBS@
 Cflags: @EXSLT_INCLUDEDIR@
diff --git a/libxslt.pc.in b/libxslt.pc.in
index ccd6e77..b85d2ca 100644
--- a/libxslt.pc.in
+++ b/libxslt.pc.in
@@ -8,6 +8,6 @@ Name: libxslt
 Version: @VERSION@
 Description: XSLT library version 2.
 Requires: libxml-2.0
-Libs: @XSLT_LIBDIR@ @XSLT_LIBS@ @EXTRA_LIBS@
+Libs: -lxslt @XSLT_LIBDIR@ @XSLT_LIBS@ @EXTRA_LIBS@
 Libs.private: @M_LIBS@
 Cflags: @XSLT_INCLUDEDIR@
