diff --git a/Makefile.am b/Makefile.am
index 14b1e40..a389462 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -40,10 +40,10 @@ libjson_cinclude_HEADERS = \
 libjson_c_la_LDFLAGS = -version-info 2:0:0 -no-undefined
 
 if ENABLE_OLDNAME_COMPAT
-libjson_la_LDFLAGS = -version-info 1:0:1 -no-undefined -ljson-c
+libjson_la_LDFLAGS = -version-info 1:0:1 -no-undefined
 
 # Temporary libjson library.  This will be removed after one release.
-libjson_la_LIBADD = -ljson-c
+libjson_la_LIBADD =
 endif
 
 
