diff --git a/Lib/distutils/sysconfig.py b/Lib/distutils/sysconfig.py
index ddb4658..119e34b 100644
--- a/Lib/distutils/sysconfig.py
+++ b/Lib/distutils/sysconfig.py
@@ -77,6 +77,8 @@ def get_python_inc(plat_specific=0, prefix=None):
     If 'prefix' is supplied, use it instead of sys.prefix or
     sys.exec_prefix -- i.e., ignore 'plat_specific'.
     """
+    if prefix is None:
+        prefix = os.environ['STAGING_INCDIR'].rstrip('include')
 
     if os.name == "posix":
         if python_build:
@@ -121,7 +121,10 @@ def get_python_lib(plat_specific=0, standard_lib=0, prefix=None):
     """
     is_default_prefix = not prefix or os.path.normpath(prefix) in ('/usr', '/usr/local')
     if prefix is None:
-        prefix = plat_specific and EXEC_PREFIX or PREFIX
+        if plat_specific:
+            prefix = os.environ['STAGING_INCDIR'].rstrip('include')
+        else:
+            prefix = PREFIX
 
     if os.name == "posix":
         libpython = os.path.join(prefix,
@@ -268,7 +272,7 @@ def get_makefile_filename():
     if python_build:
         return os.path.join(project_base, "Makefile")
     lib_dir = get_python_lib(plat_specific=1, standard_lib=1)
-    return os.path.join(get_config_var('LIBPL'), "Makefile")
+    return os.path.join(lib_dir, "config-" + os.environ['DEB_HOST_MULTIARCH'], "Makefile")
 
 
 def parse_config_h(fp, g=None):
