diff --git a/Makefile b/Makefile
index 755cc7a..e2095f3 100644
--- a/Makefile
+++ b/Makefile
@@ -116,7 +116,7 @@ distclean: clean
 
 install: all
 # -c is ignored on Linux, but required on FreeBSD
-	$(DIRINSTALL) -m 755 $(DESTDIR)$(SBINDIR) $(DESTDIR)$(IDSDIR) $(DESTDIR)$(MANDIR)/man8 $(DESTDIR)$(MANDIR)/man7
+	$(DIRINSTALL) -m 755 $(DESTDIR)$(SBINDIR) $(DESTDIR)-udeb$(SBINDIR)/lspci $(DESTDIR)$(IDSDIR) $(DESTDIR)$(MANDIR)/man8 $(DESTDIR)$(MANDIR)/man7
 	$(INSTALL) -c -m 755 lspci setpci $(pcimod-1) $(DESTDIR)$(SBINDIR)
 	$(INSTALL) -c -m 755 lspci-udeb $(DESTDIR)-udeb$(SBINDIR)/lspci
 	$(INSTALL) -c -m 755 update-pciids $(DESTDIR)$(SBINDIR)
