diff --git a/lib/configure b/lib/configure
index 419cdac..5a50fe1 100755
--- a/lib/configure
+++ b/lib/configure
@@ -53,21 +53,7 @@ cpu=`echo $host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\1/'`
 sys=`echo $host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\3/'`
 echo " $host $rel $cpu $sys"
 
-if [ "$host" = "linux--gnueabi" ]
-then
-       sys=linux
-fi
-
-if [ "$host" = "linux--uclibc" ]
-then
-       sys=linux
-fi
-
-if [ "$host" = "linux--uclibceabi" ]
-then
-       sys=linux
-fi
-
+{ echo "$host" | grep linux; } && sys=linux
 
 c=config.h
 m=config.mk
