diff --git a/Makefile b/Makefile
index 55ee973..1aed165 100644
--- a/Makefile
+++ b/Makefile
@@ -35,7 +35,7 @@ SHAREDIR=$(PREFIX)/share
 IDSDIR=$(SHAREDIR)
 MANDIR:=$(shell if [ -d $(PREFIX)/share/man ] ; then echo $(PREFIX)/share/man ; else echo $(PREFIX)/man ; fi)
 INCDIR=$(PREFIX)/include
-LIBDIR=$(PREFIX)/lib
+LIBDIR=$(libdir)
 PKGCFDIR=$(LIBDIR)/pkgconfig
 
 # Commands
@@ -98,7 +98,7 @@ example: example.o lib/$(PCILIB_DEV)
 example.o: example.c $(PCIINC)
 
 %: %.o
-	$(CC) $(LDFLAGS) $(TARGET_ARCH) $^ $(LIB_LDLIBS) -o $@
+	$(CC) $(LDFLAGS) $(TARGET_ARCH) $^ $(LIB_LDLIBS) $(LDLIBS) -o $@
 
 lspci-udeb: lspci.o
 	$(CC) $(LDFLAGS) $(TARGET_ARCH) $^ $(LDLIBS) -lz -lresolv -o $@
