diff --git a/Makefile.PL b/Makefile.PL
index a41c4ce..9cfb1b9 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -17,9 +17,9 @@ if ($Config::Config{myuname} =~ /^Win(32|64) strawberry-perl\b/) {
     libs ["eay$1"];
 }
 else {
-    inc '-I/usr/include/openssl -I/usr/local/include/ssl -I/usr/local/ssl/include';
+    inc '-I##STAGING_INCDIR##/openssl';
 
-    libs '-L/usr/lib -L/usr/local/lib -L/usr/local/ssl/lib -lcrypto';
+    libs '-L##STAGING_LIBDIR## -lcrypto';
 
     if ($Config::Config{myuname} =~ /darwin/i) {
       cc_optimize_flags('-O2 -g -Wall -Werror -Wno-deprecated-declarations');
