diff --git a/configure.ac b/configure.ac
index cbe90e0..74eedaf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -8,12 +8,6 @@ AC_CANONICAL_TARGET
 AM_INIT_AUTOMAKE([foreign no-define])
 dnl AM_MAINTAINER_MODE
 
-AC_CHECK_FILE(/usr/X/include/X11/X.h,
-  AC_PREFIX_DEFAULT(/usr/X),
-  AC_PREFIX_DEFAULT(/usr))
-AC_CHECK_FILE(/usr/X11R6/include/X11/X.h,
-  AC_PREFIX_DEFAULT(/usr/X11R6),
-  AC_PREFIX_DEFAULT(/usr))
 
 dnl The version of the Motif libraries
 CURRENT=4
