diff --git a/configure.in b/configure.in
index 440b3ca..118f94a 100644
--- a/configure.in
+++ b/configure.in
@@ -328,14 +328,12 @@ AC_MSG_RESULT($LibDirSuffix)
 case $libdir in
   */*${LibDirSuffix})	: Cool ;;
   *)			: Uh Oh... 
-	libdir=`dirname $libdir`/$LibDirSuffix
-	AC_MSG_WARN([Overriding libdir to: $libdir]);;
+	AC_MSG_WARN([Prevent overriding]);;
 esac
 case $libexecdir in
   */$LibDirSuffix)	: Cool ;;
   *)			: Uh Oh... 
-	libexecdir=`dirname $libexecdir`/$LibDirSuffix
-	AC_MSG_WARN([Overriding libexecdir to: $libexecdir]);;
+	AC_MSG_WARN([Prevent overriding]);;
 esac
 
 for j in exec_prefix bindir sbindir datadir sysconfdir localstatedir \
@@ -385,15 +383,8 @@ then
 	sleep $SNOOZING
 fi
 
-AC_CHECK_HEADERS(heartbeat/glue_config.h)
-GLUE_HEADER=none
-if test "$ac_cv_header_heartbeat_glue_config_h" = "yes";  then
-   GLUE_HEADER=heartbeat/glue_config.h
-else
-   AC_MSG_FAILURE(Core development headers were not found)
-fi
-
-CFLAGS="$CFLAGS -I${prefix}/include/heartbeat"
+GLUE_HEADER=heartbeat/glue_config.h
+CFLAGS="$CFLAGS -I${STAGING_DIR_TARGET}${prefix}/include/heartbeat"
 AC_CHECK_HEADERS(glue_config.h)
 if test "$ac_cv_header_glue_config_h" != "yes";  then
    AC_MSG_FAILURE(Core development headers were not found)
@@ -418,7 +409,7 @@ AC_SUBST(HA_NOARCHDATAHBDIR)
 AC_DEFINE_UNQUOTED(HA_NOARCHDATAHBDIR,"$HA_NOARCHDATAHBDIR", $HB_PKG noarch data directory)
 
 
-HA_LIBHBDIR=`extract_header_define $GLUE_HEADER HA_LIBHBDIR`
+HA_LIBHBDIR='/usr/lib/heartbeat'
 AC_SUBST(HA_LIBHBDIR)
 if test x"$HA_LIBHBDIR" = x ; then
 # see rhbz#805147 and
@@ -429,7 +420,7 @@ if test x"$HA_LIBHBDIR" = x ; then
 	AC_DEFINE_UNQUOTED(HA_LIBHBDIR,"$HA_LIBHBDIR", lib and daemon directory)
 fi
 
-HA_VARRUNDIR=`extract_header_define $GLUE_HEADER HA_VARRUNDIR`
+HA_VARRUNDIR='/var/run'
 AC_SUBST(HA_VARRUNDIR)
 
 HA_VARRUNHBDIR="$HA_VARRUNDIR/${HB_PKG}"
@@ -441,10 +432,10 @@ AC_SUBST(HA_VARRUNHBRSCDIR)
 HA_VARLIBDIR="${localstatedir}/lib"
 AC_SUBST(HA_VARLIBDIR)
 
-HA_VARLIBHBDIR=`extract_header_define $GLUE_HEADER HA_VARLIBHBDIR`
+HA_VARLIBHBDIR='/var/lib/heartbeat'
 AC_SUBST(HA_VARLIBHBDIR)
 
-HA_COREDIR=`extract_header_define $GLUE_HEADER HA_COREDIR`
+HA_COREDIR='/var/lib/heartbeat/cores'
 AC_SUBST(HA_COREDIR)
 
 base_includedir="${includedir}"
@@ -1749,10 +1740,10 @@ then
   IP2UTIL=/sbin/ip
 fi
 
-HA_CCMUSER=`extract_header_define $GLUE_HEADER HA_CCMUSER`
+HA_CCMUSER='hacluster'
 AC_SUBST(HA_CCMUSER)
 
-HA_APIGROUP=`extract_header_define $GLUE_HEADER HA_APIGROUP`
+HA_APIGROUP='haclient'
 AC_SUBST(HA_APIGROUP)
 
 if
@@ -1997,24 +1988,9 @@ AC_CHECK_HEADERS(evs.h, , [openais_installed="no"],)
 AC_CHECK_LIB(evs, evs_initialize , , [openais_installed="no"])
 AM_CONDITIONAL(BUILD_OPENAIS_MODULE, test "x${openais_installed}" = "xyes")
 
-dnl check if there are getpid() inconsistency
-dnl	Note: reduce LIBS; in particular, ltdl can cause confusion.
-dnl	Possibly better:  move 'LIBS="$LIBS -lltdl"' from above to beyond here.
-dnl
-AC_MSG_CHECKING(for getpid() consistency in multi-process/threads program)
-ac_save_LIBS=$LIBS
-LIBS="-lpthread"
-AC_TRY_RUN(`cat $srcdir/config/pidtest.c`, 
-AC_MSG_RESULT(ok), 
-[AC_MSG_RESULT(fail); AC_DEFINE(GETPID_INCONSISTENT, 1 , [pid inconsistent])],)
-LIBS=$ac_save_LIBS
-
 dnl check byte order
 AC_MSG_CHECKING(for byteorder)
-AC_TRY_RUN(`cat $srcdir/config/byteorder_test.c`,
-[AC_MSG_RESULT(little-endian); AC_DEFINE(CONFIG_LITTLE_ENDIAN, 1, [little-endian])],
-[AC_MSG_RESULT(big-endian); AC_DEFINE(CONFIG_BIG_ENDIAN, 1, [big-endian])],)
-
+CROSS_ENDIAN_INFO
 
 dnl
 dnl Lex and yacc can't be trusted to produce code that won't produce
@@ -2167,19 +2143,19 @@ AM_CONDITIONAL(BUILD_DRBD_OUTDATE_PEER, test "x${enable_dopd}" != "xno")
 hb_libdir="${libdir}/${HB_PKG}"
 AC_SUBST(hb_libdir)
 
-HA_PLUGIN_DIR=`extract_header_define $GLUE_HEADER HA_PLUGIN_DIR`
+HA_PLUGIN_DIR='CROSS_LIBDIR/heartbeat/plugins'
 AC_SUBST(HA_PLUGIN_DIR)
 
-HB_RA_DIR=`extract_header_define $GLUE_HEADER HB_RA_DIR`
+HB_RA_DIR='/etc/ha.d/resource.d/'
 AC_SUBST(HB_RA_DIR)
 
-OCF_ROOT_DIR=`extract_header_define $GLUE_HEADER OCF_ROOT_DIR`
+OCF_ROOT_DIR='CROSS_LIBDIR/ocf'
 AC_SUBST(OCF_ROOT_DIR)
 
-OCF_RA_DIR=`extract_header_define $GLUE_HEADER OCF_RA_DIR`
+OCF_RA_DIR='CROSS_LIBDIR/ocf/resource.d/'
 AC_SUBST(OCF_RA_DIR)
 
-LSB_RA_DIR=`extract_header_define $GLUE_HEADER LSB_RA_DIR`
+LSB_RA_DIR='/etc/init.d'
 AC_SUBST(LSB_RA_DIR)
 
 AC_ARG_ENABLE([valgrind],
@@ -2461,7 +2437,7 @@ AC_MSG_RESULT([  Mangled CFLAGS           = "${CFLAGS}"])
 AC_MSG_RESULT([  Libraries                = "${LIBS}"])
 AC_MSG_RESULT([  RPATH enabled            = "${enable_rpath}"])
 AC_MSG_RESULT([  Distro-style RPMs        = "${enable_distro_rpm}"])
-
+AC_SUBST(lt_sysroot)
 AC_MSG_RESULT([  ])
 AC_MSG_RESULT([Note: If you use the 'make install' method for installation you])
 AC_MSG_RESULT([also need to adjust '/etc/passwd' and '/etc/group' manually.])
