diff --git a/Makefile.am b/Makefile.am
index a2ed163..64e3b81 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -54,7 +54,7 @@ hauser = 		@HA_CCMUSER@
 
 # 'buildtools' must be first in the list: tools to aid building of remainder.
 SUBDIRS	= buildtools $(LIBLTDL_DIR) pkg port replace include lib	\
-          heartbeat membership telecom resources tools doc cts  \
+          heartbeat membership telecom resources tools cts  \
           config contrib
 
 HANDY_DOCS = doc/ChangeLog doc/GettingStarted.html doc/DirectoryMap.txt
