diff --git a/elf/dl-cache.c b/elf/dl-cache.c                                    
index 1966535..887aacd 100644                                                   
--- a/elf/dl-cache.c                                                            
+++ b/elf/dl-cache.c                                                            
@@ -20,7 +20,7 @@                                                               
 #include <unistd.h>                                                            
 //#include "../sysdeps/generic/ldsodefs.h"                                     
 #include <sys/mman.h>                                                          
-#include <dl-cache.h>                                                          
+#include "../sysdeps/generic/dl-cache.h"                                       
 #include <dl-procinfo.h>                                                       
 #include <stdint.h>                                                            
 //#include "_itoa.h"
diff --git a/elf/dl-cache.c b/elf/dl-cache.c                                    
index 887aacd..51d33c2 100644                                                   
--- a/elf/dl-cache.c                                                            
+++ b/elf/dl-cache.c                                                            
@@ -21,7 +21,7 @@                                                               
 //#include "../sysdeps/generic/ldsodefs.h"                                     
 #include <sys/mman.h>                                                          
 #include "../sysdeps/generic/dl-cache.h"                                       
-#include <dl-procinfo.h>                                                       
+#include "../sysdeps/generic/dl-procinfo.h"                                    
 #include <stdint.h>                                                            
 //#include "_itoa.h"
