diff --git a/aclocal.m4 b/aclocal.m4
index bc48586..bc09340 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -1,7 +1,8 @@
 dnl We require that everyone use exactly the same Autoconf version so that
 dnl the internal functions defined and used by the main configure script
 dnl match those expected by the fragments.
-m4_define([GLIBC_AUTOCONF_VERSION], [2.68])
+m4_define([GLIBC_AUTOCONF_VERSION],
+      m4_esyscmd([echo -n `autoconf --version | grep -oe "[0-9]*\.[.0-9]\+"`]))
 m4_if(m4_defn([AC_AUTOCONF_VERSION]), GLIBC_AUTOCONF_VERSION, [],
       [m4_fatal(m4_flatten(
 Exactly version GLIBC_AUTOCONF_VERSION of Autoconf is required but you have
