diff --git a/appl/login/Makefile.am b/appl/login/Makefile.am
index 3a91b85..73ecb29 100644
--- a/appl/login/Makefile.am
+++ b/appl/login/Makefile.am
@@ -34,7 +34,7 @@ LDADD = $(LIB_otp) \
 	$(DBLIB)
 
 $(srcdir)/login-protos.h: $(login_SOURCES)
-	cd $(srcdir); perl ../../cf/make-proto.pl -o login-protos.h -q -P comment $(login_SOURCES) || rm -f login-protos.h
+	cd $(srcdir); ##perl## ../../cf/make-proto.pl -o login-protos.h -q -P comment $(login_SOURCES) || rm -f login-protos.h
 
 $(login_OBJECTS): $(srcdir)/login-protos.h
 
diff --git a/cf/Makefile.am.common b/cf/Makefile.am.common
index c0cddf6..28e1ab8 100644
--- a/cf/Makefile.am.common
+++ b/cf/Makefile.am.common
@@ -2,7 +2,7 @@
 
 SUFFIXES = .et .h .pc.in .pc
 
-DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)/include -I$(top_srcdir)/include
+DEFAULT_INCLUDES = -I${STAGING_INCDIR} -I. -I$(srcdir) -I$(top_builddir)/include -I$(top_srcdir)/include
 
 AM_CPPFLAGS = $(INCLUDES_roken)
 
diff --git a/cf/check-compile-et.m4 b/cf/check-compile-et.m4
index 56a9f74..1264fde 100644
--- a/cf/check-compile-et.m4
+++ b/cf/check-compile-et.m4
@@ -95,7 +95,7 @@ elif test "${krb_cv_com_err}" = "cross"; then
     COMPILE_ET="${ac_cv_prog_COMPILE_ET}"
     localcomerr=yes
 else
-    COMPILE_ET="\$(top_builddir)/lib/com_err/compile_et"
+    COMPILE_ET="##compile_et##compile_et"
     DIR_com_err="com_err"
     LIB_com_err="\$(top_builddir)/lib/com_err/libcom_err.la"
     LIB_com_err_a="\$(top_builddir)/lib/com_err/.libs/libcom_err.a"
diff --git a/configure.ac b/configure.ac
index 945c614..e94bcca 100644
--- a/configure.ac
+++ b/configure.ac
@@ -91,7 +91,7 @@ else
    ASN1_COMPILE_DEP=
    SLC_DEP=
 
-   ac_cv_prog_COMPILE_ET=${with_cross_tools}compile_et
+   ac_cv_prog_COMPILE_ET=##compile_et##compile_et
 
 fi
 
diff --git a/kcm/Makefile.am b/kcm/Makefile.am
index 71f63c4..a6241c0 100644
--- a/kcm/Makefile.am
+++ b/kcm/Makefile.am
@@ -26,7 +26,7 @@ kcm_SOURCES =		\
 noinst_HEADERS = $(srcdir)/kcm-protos.h
 
 $(srcdir)/kcm-protos.h: $(kcm_SOURCES)
-	cd $(srcdir); perl ../cf/make-proto.pl -o kcm-protos.h -q -P comment $(kcm_SOURCES) || rm -f kcm-protos.h
+	cd $(srcdir); ##perl## ../cf/make-proto.pl -o kcm-protos.h -q -P comment $(kcm_SOURCES) || rm -f kcm-protos.h
 
 $(kcm_OBJECTS): $(srcdir)/kcm-protos.h
 
diff --git a/kdc/Makefile.am b/kdc/Makefile.am
index 085346f..49a66d3 100644
--- a/kdc/Makefile.am
+++ b/kdc/Makefile.am
@@ -73,10 +73,10 @@ endif
 $(libkdc_la_OBJECTS): $(srcdir)/version-script.map
 
 $(srcdir)/kdc-protos.h: $(libkdc_la_SOURCES)
-	cd $(srcdir) && perl ../cf/make-proto.pl -q -P comment -o kdc-protos.h $(libkdc_la_SOURCES) || rm -f kdc-protos.h
+	cd $(srcdir) && ##perl## ../cf/make-proto.pl -q -P comment -o kdc-protos.h $(libkdc_la_SOURCES) || rm -f kdc-protos.h
 
 $(srcdir)/kdc-private.h: $(libkdc_la_SOURCES)
-	cd $(srcdir) && perl ../cf/make-proto.pl -q -P comment -p kdc-private.h $(libkdc_la_SOURCES) || rm -f kdc-private.h
+	cd $(srcdir) && ##perl## ../cf/make-proto.pl -q -P comment -p kdc-private.h $(libkdc_la_SOURCES) || rm -f kdc-private.h
 
 
 hprop_LDADD = \
diff --git a/lib/asn1/Makefile.am b/lib/asn1/Makefile.am
index f2e6f1c..827e523 100644
--- a/lib/asn1/Makefile.am
+++ b/lib/asn1/Makefile.am
@@ -274,7 +274,7 @@ ALL_OBJECTS += $(check_template_OBJECTS)
 $(ALL_OBJECTS): $(DER_PROTOS) asn1_err.h
 
 $(srcdir)/der-protos.h: $(dist_libasn1base_la_SOURCES)
-	cd $(srcdir) && perl ../../cf/make-proto.pl -q -P comment -o der-protos.h $(dist_libasn1base_la_SOURCES) || rm -f der-protos.h
+	cd $(srcdir) && ##perl## ../../cf/make-proto.pl -q -P comment -o der-protos.h $(dist_libasn1base_la_SOURCES) || rm -f der-protos.h
 
 $(srcdir)/der-private.h: $(dist_libasn1base_la_SOURCES)
-	cd $(srcdir) && perl ../../cf/make-proto.pl -q -P comment -p der-private.h $(dist_libasn1base_la_SOURCES) || rm -f der-private.h
+	cd $(srcdir) && ##perl## ../../cf/make-proto.pl -q -P comment -p der-private.h $(dist_libasn1base_la_SOURCES) || rm -f der-private.h
diff --git a/lib/gssapi/Makefile.am b/lib/gssapi/Makefile.am
index 074ee78..36e63dd 100644
--- a/lib/gssapi/Makefile.am
+++ b/lib/gssapi/Makefile.am
@@ -199,7 +199,7 @@ ntlmsrc = \
 	ntlm/kdc.c
 
 $(srcdir)/ntlm/ntlm-private.h: $(ntlmsrc)
-	cd $(srcdir) && perl ../../cf/make-proto.pl -q -P comment -p ntlm/ntlm-private.h $(ntlmsrc) || rm -f ntlm/ntlm-private.h
+	cd $(srcdir) && ##perl## ../../cf/make-proto.pl -q -P comment -p ntlm/ntlm-private.h $(ntlmsrc) || rm -f ntlm/ntlm-private.h
 
 dist_libgssapi_la_SOURCES  = \
 	$(krb5src) \
@@ -288,10 +288,10 @@ gssapi_asn1_files: $(ASN1_COMPILE_DEP) $(srcdir)/mech/gssapi.asn1
 	$(ASN1_COMPILE) $(srcdir)/mech/gssapi.asn1 gssapi_asn1
 
 $(srcdir)/krb5/gsskrb5-private.h:
-	cd $(srcdir) && perl ../../cf/make-proto.pl -q -P comment -p krb5/gsskrb5-private.h $(krb5src) || rm -f krb5/gsskrb5-private.h
+	cd $(srcdir) && ##perl## ../../cf/make-proto.pl -q -P comment -p krb5/gsskrb5-private.h $(krb5src) || rm -f krb5/gsskrb5-private.h
 
 $(srcdir)/spnego/spnego-private.h:
-	cd $(srcdir) && perl ../../cf/make-proto.pl -q -P comment -p spnego/spnego-private.h $(spnegosrc) || rm -f spnego/spnego-private.h
+	cd $(srcdir) && ##perl## ../../cf/make-proto.pl -q -P comment -p spnego/spnego-private.h $(spnegosrc) || rm -f spnego/spnego-private.h
 
 
 TESTS = test_oid test_names test_cfx
@@ -352,5 +352,5 @@ gkrb5_err.h gkrb5_err.c: $(srcdir)/krb5/gkrb5_err.et
 	$(COMPILE_ET) $(srcdir)/krb5/gkrb5_err.et
 
 $(srcdir)/gssapi/gssapi_oid.h $(srcdir)/mech/gss_oid.c:
-	perl $(srcdir)/gen-oid.pl -b base -h $(srcdir)/oid.txt > $(srcdir)/gssapi/gssapi_oid.h
-	perl $(srcdir)/gen-oid.pl -b base $(srcdir)/oid.txt > $(srcdir)/mech/gss_oid.c
+	##perl## $(srcdir)/gen-oid.pl -b base -h $(srcdir)/oid.txt > $(srcdir)/gssapi/gssapi_oid.h
+	##perl## $(srcdir)/gen-oid.pl -b base $(srcdir)/oid.txt > $(srcdir)/mech/gss_oid.c
diff --git a/lib/hdb/Makefile.am b/lib/hdb/Makefile.am
index a0590d6..1979958 100644
--- a/lib/hdb/Makefile.am
+++ b/lib/hdb/Makefile.am
@@ -126,10 +126,10 @@ $(ALL_OBJECTS): $(HDB_PROTOS)
 $(libhdb_la_OBJECTS): hdb_asn1.h hdb_asn1-priv.h hdb_err.h
 
 $(srcdir)/hdb-protos.h: $(dist_libhdb_la_SOURCES)
-	cd $(srcdir); perl ../../cf/make-proto.pl -q -P comment -o hdb-protos.h $(dist_libhdb_la_SOURCES) || rm -f hdb-protos.h
+	cd $(srcdir); ##perl## ../../cf/make-proto.pl -q -P comment -o hdb-protos.h $(dist_libhdb_la_SOURCES) || rm -f hdb-protos.h
 
 $(srcdir)/hdb-private.h: $(dist_libhdb_la_SOURCES)
-	cd $(srcdir); perl ../../cf/make-proto.pl -q -P comment -p hdb-private.h $(dist_libhdb_la_SOURCES) || rm -f hdb-private.h
+	cd $(srcdir); ##perl## ../../cf/make-proto.pl -q -P comment -p hdb-private.h $(dist_libhdb_la_SOURCES) || rm -f hdb-private.h
 
 $(gen_files_hdb) hdb_asn1.hx hdb_asn1-priv.hx: hdb_asn1_files
 
diff --git a/lib/hx509/Makefile.am b/lib/hx509/Makefile.am
index e4b33f3..cb7e79f 100644
--- a/lib/hx509/Makefile.am
+++ b/lib/hx509/Makefile.am
@@ -148,10 +148,10 @@ $(libhx509_la_OBJECTS): $(srcdir)/hx_locl.h
 $(libhx509_la_OBJECTS): ocsp_asn1.h pkcs10_asn1.h
 
 $(srcdir)/hx509-protos.h: $(dist_libhx509_la_SOURCES)
-	$(heim_verbose)cd $(srcdir) && perl ../../cf/make-proto.pl -R '^(_|^C)' -E HX509_LIB -q -P comment -o hx509-protos.h $(dist_libhx509_la_SOURCES) || rm -f hx509-protos.h
+	$(heim_verbose)cd $(srcdir) && ##perl## ../../cf/make-proto.pl -R '^(_|^C)' -E HX509_LIB -q -P comment -o hx509-protos.h $(dist_libhx509_la_SOURCES) || rm -f hx509-protos.h
 
 $(srcdir)/hx509-private.h: $(dist_libhx509_la_SOURCES)
-	$(heim_verbose)cd $(srcdir) && perl ../../cf/make-proto.pl -q -P comment -p hx509-private.h $(dist_libhx509_la_SOURCES) || rm -f hx509-private.h
+	$(heim_verbose)cd $(srcdir) && ##perl## ../../cf/make-proto.pl -q -P comment -p hx509-private.h $(dist_libhx509_la_SOURCES) || rm -f hx509-private.h
 
 bin_PROGRAMS = hxtool
 
diff --git a/lib/kadm5/Makefile.am b/lib/kadm5/Makefile.am
index f8be3b1..b5292be 100644
--- a/lib/kadm5/Makefile.am
+++ b/lib/kadm5/Makefile.am
@@ -181,14 +181,14 @@ KADM5_PROTOS_SRCS += $(dist_libkadm5srv_la_SOURCES)
 
 proto_opts = -q -R '^(_|kadm5_c_|kadm5_s_|kadm5_log)' -P comment
 $(srcdir)/kadm5-protos.h: $(KADM5_PROTOS_SRCS)
-	cd $(srcdir); perl ../../cf/make-proto.pl $(proto_opts) \
+	cd $(srcdir); ##perl## ../../cf/make-proto.pl $(proto_opts) \
 		-o kadm5-protos.h \
 		$(dist_libkadm5clnt_la_SOURCES) \
 		$(dist_libkadm5srv_la_SOURCES) \
 		|| rm -f kadm5-protos.h
 
 $(srcdir)/kadm5-private.h: $(KADM5_PROTOS_SRCS)
-	cd $(srcdir); perl ../../cf/make-proto.pl $(proto_opts) \
+	cd $(srcdir); ##perl## ../../cf/make-proto.pl $(proto_opts) \
 		-p kadm5-private.h \
 		$(dist_libkadm5clnt_la_SOURCES) \
 		$(dist_libkadm5srv_la_SOURCES) \
diff --git a/lib/krb5/Makefile.am b/lib/krb5/Makefile.am
index a2058e4..d4d4aa6 100644
--- a/lib/krb5/Makefile.am
+++ b/lib/krb5/Makefile.am
@@ -308,10 +308,10 @@ headerdeps = $(dist_libkrb5_la_SOURCES)
 endif
 
 $(srcdir)/krb5-protos.h: $(headerdeps)
-	@cd $(srcdir) && perl ../../cf/make-proto.pl -E KRB5_LIB -q -P comment -o krb5-protos.h $(dist_libkrb5_la_SOURCES) || rm -f krb5-protos.h
+	@cd $(srcdir) && ##perl## ../../cf/make-proto.pl -E KRB5_LIB -q -P comment -o krb5-protos.h $(dist_libkrb5_la_SOURCES) || rm -f krb5-protos.h
 
 $(srcdir)/krb5-private.h: $(headerdeps)
-	@cd $(srcdir) && perl ../../cf/make-proto.pl -q -P comment -p krb5-private.h $(dist_libkrb5_la_SOURCES) || rm -f krb5-private.h
+	@cd $(srcdir) && ##perl## ../../cf/make-proto.pl -q -P comment -p krb5-private.h $(dist_libkrb5_la_SOURCES) || rm -f krb5-private.h
 
 man_MANS =					\
 	kerberos.8				\
diff --git a/lib/ntlm/Makefile.am b/lib/ntlm/Makefile.am
index 53ed60d..6c9954f 100644
--- a/lib/ntlm/Makefile.am
+++ b/lib/ntlm/Makefile.am
@@ -27,7 +27,7 @@ libheimntlm_la_LIBADD = \
 	$(LIBADD_roken)
 
 $(srcdir)/heimntlm-protos.h: $(dist_libheimntlm_la_SOURCES)
-	cd $(srcdir) && perl ../../cf/make-proto.pl -q -P comment -o heimntlm-protos.h $(dist_libheimntlm_la_SOURCES) || rm -f heimntlm-protos.h
+	cd $(srcdir) && ##perl## ../../cf/make-proto.pl -q -P comment -o heimntlm-protos.h $(dist_libheimntlm_la_SOURCES) || rm -f heimntlm-protos.h
 
 $(libheimntlm_la_OBJECTS): $(srcdir)/heimntlm-protos.h ntlm_err.h
 
diff --git a/cf/check-compile-et.m4 b/cf/check-compile-et.m4
index 22efc77..b353928 100644
--- a/cf/check-compile-et.m4
+++ b/cf/check-compile-et.m4
@@ -24,8 +24,8 @@ EOF
 if ${COMPILE_ET} conftest_et.et >/dev/null 2>&1; then
   dnl XXX Some systems have <et/com_err.h>.
   save_CPPFLAGS="${CPPFLAGS}"
-  if test -d "/usr/include/et"; then
-    CPPFLAGS="-I/usr/include/et ${CPPFLAGS}"
+  if test -d "##STAGING_INCDIR##/et"; then
+    CPPFLAGS="-I##STAGING_INCDIR##/et ${CPPFLAGS}"
   fi
   dnl Check that the `prefix' and `index' directives were honored.
   AC_RUN_IFELSE([
