diff --git a/bin/tests/dst/Makefile.in b/bin/tests/dst/Makefile.in
index de66315..f1feaaa 100644
--- a/bin/tests/dst/Makefile.in
+++ b/bin/tests/dst/Makefile.in
@@ -58,11 +58,11 @@ gsstest@EXEEXT@: gsstest.@O@ ${DEPLIBS}
 		gsstest.@O@ ${LIBS}
 
 test: t_dst@EXEEXT@ randomfile
-	../../tools/genrandom@EXEEXT@ 100 randomfile
+	genrandom@EXEEXT@ 100 randomfile
 	-@ ./t_dst@EXEEXT@ -q 1800 -a
 
 randomfile:
-	../../tools/genrandom@EXEEXT@ 100 randomfile
+	genrandom@EXEEXT@ 100 randomfile
 
 clean distclean::
 	rm -f ${TARGETS} randomfile
