diff --git a/lib/export/dns/Makefile.in b/lib/export/dns/Makefile.in
index 1e4540f..708c9c2 100644
--- a/lib/export/dns/Makefile.in
+++ b/lib/export/dns/Makefile.in
@@ -103,7 +103,7 @@ SUBDIRS =	include
 TARGETS =	include/dns/enumtype.h include/dns/enumclass.h \
 		include/dns/rdatastruct.h timestamp
 
-DEPENDEXTRA =	./gen -F include/dns/rdatastruct.h \
+DEPENDEXTRA =	gen-lib-export-dns -F include/dns/rdatastruct.h \
 		-s ${srcdir} -d >> Makefile ;
 
 @BIND9_MAKE_RULES@
@@ -152,20 +152,20 @@ include: include/dns/enumtype.h include/dns/enumclass.h \
 rdata.@O@: code.h
 
 include/dns/enumtype.h: gen
-	./gen -s ${srcdir} -t > $@
+	gen-lib-export-dns -s ${srcdir} -t > $@
 
 include/dns/enumclass.h: gen
-	./gen -s ${srcdir} -c > $@
+	gen-lib-export-dns -s ${srcdir} -c > $@
 
 include/dns/rdatastruct.h: gen \
 		${srcdir}/rdata/rdatastructpre.h \
 		${srcdir}/rdata/rdatastructsuf.h
-	./gen -s ${srcdir} -i \
+	gen-lib-export-dns -s ${srcdir} -i \
 		-P ${srcdir}/rdata/rdatastructpre.h \
 		-S ${srcdir}/rdata/rdatastructsuf.h > $@
 
 code.h:	gen
-	./gen -s ${srcdir} > code.h
+	gen-lib-export-dns -s ${srcdir} > code.h
 
 gen: ${srcdir}/gen.c
 	${CC} ${ALL_CFLAGS} ${LDFLAGS} -o $@ ${srcdir}/gen.c ${LIBS}
diff --git a/lib/dns/Makefile.in b/lib/dns/Makefile.in
index 5bf850a..25a3a5f 100644
--- a/lib/dns/Makefile.in
+++ b/lib/dns/Makefile.in
@@ -112,7 +112,7 @@ TARGETS =	include/dns/enumtype.h include/dns/enumclass.h \
 		include/dns/rdatastruct.h timestamp
 TESTDIRS =	@UNITTESTS@
 
-DEPENDEXTRA =	./gen -F include/dns/rdatastruct.h \
+DEPENDEXTRA =	gen-lib-dns -F include/dns/rdatastruct.h \
 		-s ${srcdir} -d >> Makefile ;
 
 @BIND9_MAKE_RULES@
@@ -166,20 +166,20 @@ include: include/dns/enumtype.h include/dns/enumclass.h \
 rdata.@O@: code.h
 
 include/dns/enumtype.h: gen
-	./gen -s ${srcdir} -t > $@
+	gen-lib-dns -s ${srcdir} -t > $@
 
 include/dns/enumclass.h: gen
-	./gen -s ${srcdir} -c > $@
+	gen-lib-dns -s ${srcdir} -c > $@
 
 include/dns/rdatastruct.h: gen \
 		${srcdir}/rdata/rdatastructpre.h \
 		${srcdir}/rdata/rdatastructsuf.h
-	./gen -s ${srcdir} -i \
+	gen-lib-dns -s ${srcdir} -i \
 		-P ${srcdir}/rdata/rdatastructpre.h \
 		-S ${srcdir}/rdata/rdatastructsuf.h > $@
 
 code.h:	gen
-	./gen -s ${srcdir} > code.h
+	gen-lib-dns -s ${srcdir} > code.h
 
 gen: gen.c
 	${BUILD_CC} ${BUILD_CFLAGS} -I${top_srcdir}/lib/isc/include \
