diff --git a/bin/tests/Makefile.in b/bin/tests/Makefile.in
index bc040a3..c61fd46 100644
--- a/bin/tests/Makefile.in
+++ b/bin/tests/Makefile.in
@@ -41,7 +41,7 @@ LWRESDEPLIBS =	../../lib/lwres/liblwres.@A@
 
 LIBS =		@LIBS@
 
-SUBDIRS =	atomic db dst master mem hashes names net rbt resolver \
+SUBDIRS =	atomic db master mem hashes names net rbt resolver \
 		sockaddr tasks timers system
 
 # Test programs that are built by default:
