From 58292175115bae7a4c3600d3022e59ea57213025 Mon Sep 17 00:00:00 2001
From: Jianchuan Wang <jianchuan.wang@windriver.com>
Date: Mon, 12 Jan 2015 15:53:23 +0800
Subject: [PATCH] Modify the Makefile for cross compile.

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Jianchuan Wang <jianchuan.wang@windriver.com>
---
 Makefile         | 23 ++++++-----------------
 libipvs/Makefile |  3 +--
 2 files changed, 7 insertions(+), 19 deletions(-)

diff --git a/libipvs/Makefile b/libipvs/Makefile
index a527a7f..b7300ec 100644
--- a/libipvs/Makefile
+++ b/libipvs/Makefile
@@ -1,6 +1,5 @@
 # Makefile for libipvs
 
-CC		= gcc
 CFLAGS		= -Wall -Wunused -Wstrict-prototypes -g -fPIC
 ifneq (0,$(HAVE_NL))
 CFLAGS		+= -DLIBIPVS_USE_NL
@@ -18,7 +17,7 @@ SHARED_LIB	= libipvs.so
 all:		$(STATIC_LIB) $(SHARED_LIB)
 
 $(STATIC_LIB):	libipvs.o ip_vs_nl_policy.o
-		ar rv $@ $^
+		$(AR) rv $@ $^
 
 $(SHARED_LIB):	libipvs.o ip_vs_nl_policy.o
 		$(CC) -shared -Wl,-soname,$@ -o $@ $^
-- 
1.9.1
