diff --git a/libkmod/libkmod-signature.c b/libkmod/libkmod-signature.c
index a3ac15e..646a879 100644
--- a/libkmod/libkmod-signature.c
+++ b/libkmod/libkmod-signature.c
@@ -19,6 +19,7 @@
  */
 
 #include <endian.h>
+#include <byteswap.h>
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
@@ -121,7 +122,7 @@ bool kmod_module_signature_info(const struct kmod_file *file, struct kmod_signat
 			modsig->hash >= PKEY_HASH__LAST ||
 			modsig->id_type >= PKEY_ID_TYPE__LAST)
 		return false;
-	sig_len = be32toh(get_unaligned(&modsig->sig_len));
+	sig_len = bswap_32(get_unaligned(&modsig->sig_len));
 	if (size < (off_t)(modsig->signer_len + modsig->key_id_len + sig_len))
 		return false;
 
