diff --git a/Makefile.am b/Makefile.am
index c05921a..d32b3d3 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -16,7 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-SUBDIRS = gl lib/gl lib po src doc gltests lib/gltests tests examples	\
+SUBDIRS = gl lib/gl lib po src gltests lib/gltests tests examples	\
 	java csharp
 
 EXTRA_DIST = cfg.mk maint.mk .clcopying
