diff --git a/src/Makefile.am b/src/Makefile.am
index 9e07f71..f66cddb 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -39,8 +39,8 @@ idn_cmd.c idn_cmd.h: idn.ggo Makefile.am
 	gengetopt --unamed-opts --no-handle-version --no-handle-help \
 		--set-package="idn" \
 		--input $^ --file-name idn_cmd
-	perl -pi -e 's/\[OPTIONS\]/\[OPTION\]/g' idn_cmd.c
-	perl -pi -e 's/\[FILES\]/\[STRING\]/g' idn_cmd.c
+	perl -pi -e "s/\[OPTIONS\]/\[OPTION\]/g" idn_cmd.c
+	perl -pi -e "s/\[FILES\]/\[STRING\]/g" idn_cmd.c
 
 BUILT_SOURCES = idn_cmd.c idn_cmd.h
 MAINTAINERCLEANFILES = $(BUILT_SOURCES)
