From 526ad32fd4db8e7b78803aaa3e4e69507abcbbb2 Mon Sep 17 00:00:00 2001
From: Toshiko YOSHIDA <toshiko.yoshida@toshiba.co.jp>
Date: Thu, 5 Oct 2017 09:15:06 +0900
Subject: [PATCH] create relative symbolic link

Signed-off-by: Toshiko YOSHIDA <toshiko.yoshida@toshiba.co.jp>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c904eaf..6ae7f3b 100644
--- a/Makefile
+++ b/Makefile
@@ -161,7 +161,7 @@ endif
 	$(INSTALL) -D $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(LIBNAME)
 	$(LNS) $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(SONAME)
 	mkdir -p $(DESTDIR)$(USRLIBDIR)
-	$(LNS) $(LIBDIR)/$(SONAME) $(DESTDIR)$(USRLIBDIR)/$(DEVELLIB)
+	$(LNS) -r $(DESTDIR)$(LIBDIR)/$(SONAME) $(DESTDIR)$(USRLIBDIR)/$(DEVELLIB)
 	$(INSTALL) -D keyctl $(DESTDIR)$(BINDIR)/keyctl
 	$(INSTALL) -D request-key $(DESTDIR)$(SBINDIR)/request-key
 	$(INSTALL) -D request-key-debug.sh $(DESTDIR)$(SHAREDIR)/request-key-debug.sh
-- 
2.1.4

