From 431a6e66ffbcf5a85d72b02f2d34118d5cd51235 Mon Sep 17 00:00:00 2001
From: Toshiko YOSHIDA <toshiko.yoshida@toshiba.co.jp>
Date: Wed, 11 Oct 2017 16:45:49 +0900
Subject: [PATCH] avoid using python-native by default

Signed-off-by: Toshiko YOSHIDA <toshiko.yoshida@toshiba.co.jp>
---
 geo-replication/src/gsyncd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/geo-replication/src/gsyncd.c b/geo-replication/src/gsyncd.c
index 0830e7f..984b396 100644
--- a/geo-replication/src/gsyncd.c
+++ b/geo-replication/src/gsyncd.c
@@ -145,7 +145,7 @@ invoke_gsyncd (int argc, char **argv)
         j = 0;
         python = getenv("PYTHON");
         if(!python)
-                python = PYTHON;
+                python = "/usr/bin/python";
         nargv[j++] = python;
         nargv[j++] = GSYNCD_PREFIX"/python/syncdaemon/"GSYNCD_PY;
         for (i = 1; i < argc; i++)
-- 
2.1.4

