diff --git a/Makefile b/Makefile
index 7da6496..71b5948 100644
--- a/Makefile
+++ b/Makefile
@@ -105,13 +105,14 @@ dep:
 
 # Installation related variables and target
 SED?=sed
+PREFIX=/usr
 INSTALL_PREFIX?=$(DESTDIR)/usr
 PREFIX?=$(INSTALL_PREFIX)
 INCLUDE_PATH?=include/hiredis
 LIBRARY_PATH?=lib
 LIB_PREFIX?=$(PREFIX)/$(LIBRARY_PATH)
 PKGCONFIG_PATH?=$(LIBRARY_PATH)/pkgconfig
-CMAKE_MODULES_PATH?=share/cmake/Modules
+CMAKE_MODULES_PATH?=share/cmake-2.8/Modules
 INSTALL_INCLUDE_PATH= $(INSTALL_PREFIX)/$(INCLUDE_PATH)
 INSTALL_LIBRARY_PATH= $(INSTALL_PREFIX)/$(LIBRARY_PATH)
 INSTALL_PKGCONFIG_PATH= $(INSTALL_PREFIX)/$(PKGCONFIG_PATH)
