diff --git a/misc/mke2fs.c b/misc/mke2fs.c
index 1552c1c..62dd396 100644
--- a/misc/mke2fs.c
+++ b/misc/mke2fs.c
@@ -2938,10 +2938,11 @@ no_journal:
 		current_fs = fs;
 		root = EXT2_ROOT_INO;
 		retval = populate_fs(root, root_dir);
-		if (retval)
+		if (retval) {
 			fprintf(stderr, "%s",
-				_("\nError while populating file system"));
-		else if (!quiet)
+				_("\nError while populating file system\n"));
+			return retval;
+		} else if (!quiet)
 		        printf("%s", _("done\n"));
 	}
 
