diff --git a/tests/qemuhelptest.c b/tests/qemuhelptest.c
index 271fddc..5b5970c 100644
--- a/tests/qemuhelptest.c
+++ b/tests/qemuhelptest.c
@@ -58,7 +58,7 @@ static int testHelpStrParsing(const void *data)
         goto cleanup;
 
     if (virQEMUCapsParseHelpStr("QEMU", help, flags,
-                                &version, &is_kvm, &kvm_version, false, NULL) == -1) {
+                                &version, &is_kvm, &kvm_version, false, NULL) == -1)
         goto cleanup;
 
 # ifndef WITH_YAJL
