diff --git a/configure.ac b/configure.ac
index 5d9fee2..c0f9dcb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -152,6 +152,8 @@ AC_PROG_CC
 AM_DISABLE_STATIC
 AM_CONDITIONAL([ENABLE_STATIC], [test "$enable_static" = yes])
 
+AM_CONDITIONAL([HAVE_INTROSPECTION], [false])
+
 dnl 
 dnl Check for a working C++ compiler, but do not bail out, if none is found.
 dnl We use this for an automated test for C++ header correctness.
