diff --git a/DebianNet.pm b/DebianNet.pm
index 338f749..3ddad1a 100644
--- a/DebianNet.pm
+++ b/DebianNet.pm
@@ -163,7 +163,7 @@ sub add_service {
                 move("$new_inetdcf","$inetdcf") ||
                     die "Error installing $new_inetdcf to $inetdcf: $!\n";
                 chmod(0644, "$inetdcf");
-                &wakeup_inetd(0,$init_svc_count);
+                #&wakeup_inetd(0,$init_svc_count);
                 &printv("New service(s) added\n");
             } else {
                 &printv("No service(s) added\n");
@@ -223,7 +223,7 @@ sub remove_service {
         move("$new_inetdcf", "$inetdcf") ||
             die "Error installing $new_inetdcf to $inetdcf: $!\n";
         chmod(0644, "$inetdcf");
-        wakeup_inetd(1);
+        #wakeup_inetd(1);
         &printv("Number of service entries removed: $nlines_removed\n");
     } else {
         &printv("No service entries were removed\n");
@@ -274,7 +274,7 @@ sub disable_service {
         move("$new_inetdcf","$inetdcf") ||
             die "Error installing new $inetdcf: $!\n";
         chmod(0644, "$inetdcf");
-        wakeup_inetd(1);
+        #wakeup_inetd(1);
         &printv("Number of service entries disabled: $nlines_disabled\n");
     } else {
         &printv("No service entries were disabled\n");
@@ -311,7 +311,7 @@ sub enable_service {
         move("$new_inetdcf","$inetdcf") ||
             die "Error installing $new_inetdcf to $inetdcf: $!\n";
         chmod(0644, "$inetdcf");
-        &wakeup_inetd(0,$init_svc_count);
+        #&wakeup_inetd(0,$init_svc_count);
         &printv("Number of service entries enabled: $nlines_enabled\n");
     } else {
         &printv("No service entries were enabled\n");
diff --git a/update-inetd b/update-inetd
index c053a7c..9827676 100755
--- a/update-inetd
+++ b/update-inetd
@@ -125,9 +125,9 @@ if ($mode eq "add") {
 
 # in case of xinetd, we have no clue about whether to
 # restart/SIGHUP/noop so we always restart anyway
-if ($xinetd && !$DebianNet::called_wakeup_inetd) {
-    DebianNet::wakeup_inetd(0);
-}
+#if ($xinetd && !$DebianNet::called_wakeup_inetd) {
+#    DebianNet::wakeup_inetd(0);
+#}
 
 sub version {
     print STDERR "$0 $version\n";
