diff --git a/Makefile.am b/Makefile.am
index f8a0a3e..78d99b3 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -52,7 +52,7 @@ NOAUTO_DIRS = examples rpm doc
 # module_dirs preceedes . due to _depmod rule
 # libcr_dirs preceedes util_dirs to ensure up-to-date libs for the utils
 SUBDIRS = include $(module_dirs) $(libcr_dirs) $(util_dirs) \
-	  $(etc_dir) $(tests_dirs) $(contrib_dirs) .
+	  $(etc_dir) $(contrib_dirs) .
 # Automake has been observed to build DIST_SUBDIRS incorrectly, so we build it
 DIST_SUBDIRS = man include libcr util tests contrib etc \
 		blcr_imports cr_module vmadump4 $(NOAUTO_DIRS)
diff --git a/blcr_imports/Makefile.am b/blcr_imports/Makefile.am
index f0ced91..7359b19 100644
--- a/blcr_imports/Makefile.am
+++ b/blcr_imports/Makefile.am
@@ -15,7 +15,7 @@ echoval:
 EXTRA_DIST = license.txt module.c
 
 # Must visit . first, to keep generated source file imports.c up to date
-SUBDIRS = . kbuild
+SUBDIRS = .
 
 dist-hook:
 	rm -f $(distdir)/imports.c
diff --git a/cr_module/Makefile.am b/cr_module/Makefile.am
index e85c2e3..70fdbe8 100644
--- a/cr_module/Makefile.am
+++ b/cr_module/Makefile.am
@@ -46,4 +46,4 @@ echoval:
 .PHONY: echoval
 
 EXTRA_DIST = arch license.txt $(my_sources)
-SUBDIRS = kbuild .
+SUBDIRS = .
