diff --git a/glib/Makefile.am b/glib/Makefile.am
index 178f703..b6448b8 100644
--- a/glib/Makefile.am
+++ b/glib/Makefile.am
@@ -101,11 +101,11 @@ $(ASSEMBLY): $(build_sources) gtk-sharp.snk AssemblyInfo.cs
 	$(CSC) $(CSFLAGS) -nowarn:0169,0612,0618 -out:$(ASSEMBLY) -target:library $(references) $(build_sources)
 	$(GAPI_CDECL_INSERT)
 
-policy.%.config: $(top_builddir)/policy.config
+policy.%.$(ASSEMBLY_NAME).config: $(top_builddir)/policy.config
 	sed -e "s/@ASSEMBLY_NAME@/$(ASSEMBLY_NAME)/" -e "s/@POLICY@/$*/" $(top_builddir)/policy.config > $@
 
-$(POLICY_ASSEMBLIES) : policy.%.$(ASSEMBLY): policy.%.config gtk-sharp.snk
-	$(AL) -link:policy.$*.config -out:$@ -keyfile:gtk-sharp.snk
+$(POLICY_ASSEMBLIES) : policy.%.$(ASSEMBLY): policy.%.$(ASSEMBLY_NAME).config gtk-sharp.snk
+	$(AL) -link:policy.$*.$(ASSEMBLY_NAME).config -out:$@ -keyfile:gtk-sharp.snk
 
 install-data-local:
 	@if test -n '$(TARGET)'; then							\
diff --git a/Makefile.include b/Makefile.include
index 424b6e5..f1cf2ca 100644
--- a/Makefile.include
+++ b/Makefile.include
@@ -46,11 +46,11 @@ $(SNK): $(top_srcdir)/$(SNK)
 AssemblyInfo.cs: $(top_builddir)/AssemblyInfo.cs
 	cp $(top_builddir)/AssemblyInfo.cs .
 
-policy.%.config: $(top_builddir)/policy.config
+policy.%.$(ASSEMBLY_NAME).config: $(top_builddir)/policy.config
 	sed -e "s/@ASSEMBLY_NAME@/$(ASSEMBLY_NAME)/" -e "s/@POLICY@/$*/" $(top_builddir)/policy.config > $@
 
-$(POLICY_ASSEMBLIES) : policy.%.$(ASSEMBLY): policy.%.config $(SNK)
-	$(AL) -link:policy.$*.config -out:$@ -keyfile:$(SNK)
+$(POLICY_ASSEMBLIES) : policy.%.$(ASSEMBLY): policy.%.$(ASSEMBLY_NAME).config $(SNK)
+	$(AL) -link:policy.$*.$(ASSEMBLY_NAME).config -out:$@ -keyfile:$(SNK)
 
 build_sources = $(addprefix $(srcdir)/, $(sources)) AssemblyInfo.cs
 build_references = $(addprefix /r:, $(references)) $(MONO_CAIRO_LIBS)
diff --git a/gtkdotnet/Makefile.am b/gtkdotnet/Makefile.am
index 56615a8..1b8f6eb 100644
--- a/gtkdotnet/Makefile.am
+++ b/gtkdotnet/Makefile.am
@@ -45,11 +45,11 @@ $(ASSEMBLY): $(build_sources) $(references) gtk-sharp.snk AssemblyInfo.cs
 	$(CSC) $(CSFLAGS) -nowarn:0169,0612,0618 -out:$(ASSEMBLY) -target:library $(build_references) $(build_sources)
 	$(GAPI_CDECL_INSERT)
 
-policy.%.config: $(top_builddir)/policy.config
+policy.%.$(ASSEMBLY_NAME).config: $(top_builddir)/policy.config
 	sed -e "s/@ASSEMBLY_NAME@/$(ASSEMBLY_NAME)/" -e "s/@POLICY@/$*/" $(top_builddir)/policy.config > $@
 
-$(POLICY_ASSEMBLIES) : policy.%.$(ASSEMBLY): policy.%.config gtk-sharp.snk
-	$(AL) -link:policy.$*.config -out:$@ -keyfile:gtk-sharp.snk
+$(POLICY_ASSEMBLIES) : policy.%.$(ASSEMBLY): policy.%.$(ASSEMBLY_NAME).config gtk-sharp.snk
+	$(AL) -link:policy.$*.$(ASSEMBLY_NAME).config -out:$@ -keyfile:gtk-sharp.snk
 
 install-data-local:
 	@if test -n '$(TARGET)'; then							\
