diff --git a/configure.ac b/configure.ac
index 21cac22..62e35eb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -18,7 +18,7 @@ dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ([2.62])
 AC_INIT([gettext],
-	m4_esyscmd([build-aux/git-version-gen .tarball-version]),
+	m4_esyscmd([echo -n `grep -m1 "released" ChangeLog | grep -o [0-9.]*[0-9]`]),
 	[bug-gnu-gettext@gnu.org])
 AC_CONFIG_SRCDIR([gettext-tools/src/msgfmt.c])
 AC_CONFIG_AUX_DIR([build-aux])
diff --git a/gettext-runtime/configure.ac b/gettext-runtime/configure.ac
index 4a1788c..0681380 100644
--- a/gettext-runtime/configure.ac
+++ b/gettext-runtime/configure.ac
@@ -18,7 +18,7 @@ dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ([2.62])
 AC_INIT([gettext-runtime],
-	m4_esyscmd([../build-aux/git-version-gen ../.tarball-version]),
+	m4_esyscmd([echo -n `grep -m1 "released" ../ChangeLog | grep -o [0-9.]*[0-9]`]),
 	[bug-gnu-gettext@gnu.org])
 AC_CONFIG_SRCDIR([intl/dcigettext.c])
 AC_CONFIG_AUX_DIR([../build-aux])
diff --git a/gettext-tools/configure.ac b/gettext-tools/configure.ac
index 650bd31..18daa3a 100644
--- a/gettext-tools/configure.ac
+++ b/gettext-tools/configure.ac
@@ -18,7 +18,7 @@ dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ([2.62])
 AC_INIT([gettext-tools],
-	m4_esyscmd([../build-aux/git-version-gen ../.tarball-version]),
+	m4_esyscmd([echo -n `grep -m1 "released" ../ChangeLog | grep -o [0-9.]*[0-9]`]),
 	[bug-gnu-gettext@gnu.org])
 AC_CONFIG_SRCDIR([src/msgfmt.c])
 AC_CONFIG_AUX_DIR([../build-aux])
diff --git a/gettext-tools/examples/configure.ac b/gettext-tools/examples/configure.ac
index ff3eaf1..ed75d61 100644
--- a/gettext-tools/examples/configure.ac
+++ b/gettext-tools/examples/configure.ac
@@ -18,7 +18,7 @@ dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ([2.57])
 AC_INIT([gettext-examples],
-	m4_esyscmd([../../build-aux/git-version-gen ../../.tarball-version]),
+	m4_esyscmd([echo -n `grep -m1 "released" ../../ChangeLog | grep -o [0-9.]*[0-9]`]),
 	[bug-gnu-gettext@gnu.org])
 AC_CONFIG_SRCDIR([installpaths.in])
 AC_CONFIG_AUX_DIR([../../build-aux])
