diff --git a/Makefile b/Makefile
index 6e1edbd..6ecc325 100644
--- a/Makefile
+++ b/Makefile
@@ -9,8 +9,8 @@ endif
 .PHONY: default
 default: startup all
 
-.PHONY: headers library clean veryclean all binary program doc test update-po
-all headers library clean veryclean binary program doc manpages docbook test update-po startup dirs:
+.PHONY: headers library clean veryclean all binary program doc update-po
+all headers library clean veryclean binary program doc manpages docbook update-po startup dirs:
 	$(MAKE) -C vendor $@
 	$(MAKE) -C apt-pkg $@
 	$(MAKE) -C apt-inst $@
@@ -21,9 +21,8 @@ all headers library clean veryclean binary program doc manpages docbook test upd
 	$(MAKE) -C dselect $@
 	$(MAKE) -C doc $@
 	$(MAKE) -C po $@
-	$(MAKE) -C test $@
 
-all headers library clean veryclean binary program doc manpages docbook test update-po: startup dirs
+all headers library clean veryclean binary program doc manpages docbook update-po: startup dirs
 
 dirs: startup
 
diff --git a/configure.ac b/configure.ac
index ed129aa..76bf43b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -82,13 +82,6 @@ AC_CHECK_HEADER(db.h,
 
 LIBS="$saveLIBS"
 
-AC_LANG_PUSH([C++])
-AC_CHECK_HEADER(gtest/gtest.h,,
-     AC_MSG_ERROR([failed: I need gtest to build tests]),
-)
-AC_LANG_POP([C++])
-
-
 AC_SUBST(BDBLIB)
 
 HAVE_ZLIB=no
