diff --git a/configure.ac b/configure.ac
index e605528..ed129aa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -82,13 +82,6 @@ AC_CHECK_HEADER(db.h,
 
 LIBS="$saveLIBS"
 
-AC_CHECK_LIB(curl, curl_easy_init,
-     [AC_CHECK_HEADER(curl/curl.h,
-        curl_ok=yes,
-        curl_ok=no)],
-     AC_MSG_ERROR([failed: I need CURL due https support]),
-)
-
 AC_LANG_PUSH([C++])
 AC_CHECK_HEADER(gtest/gtest.h,,
      AC_MSG_ERROR([failed: I need gtest to build tests]),
diff --git a/methods/makefile b/methods/makefile
index 6b77812..381e08a 100644
--- a/methods/makefile
+++ b/methods/makefile
@@ -51,13 +51,6 @@ LIB_MAKES = apt-pkg/makefile
 SOURCE = http.cc http_main.cc rfc2553emu.cc connect.cc server.cc
 include $(PROGRAM_H)
 
-# The https method
-PROGRAM=https
-SLIBS = -lapt-pkg -lcurl $(INTLLIBS)
-LIB_MAKES = apt-pkg/makefile
-SOURCE = https.cc server.cc
-include $(PROGRAM_H)
-
 # The ftp method
 PROGRAM=ftp
 SLIBS = -lapt-pkg $(SOCKETLIBS) $(INTLLIBS)
