diff --git a/apt-pkg/packagemanager.cc b/apt-pkg/packagemanager.cc
index 393f836..405f480 100644
--- a/apt-pkg/packagemanager.cc
+++ b/apt-pkg/packagemanager.cc
@@ -905,10 +905,12 @@ bool pkgPackageManager::SmartUnPack(PkgIterator Pkg, bool const Immediate, int c
       return false;
 
    if (Immediate == true) {
+#if 0
       // Perform immedate configuration of the package. 
          if (SmartConfigure(Pkg, Depth + 1) == false)
             _error->Error(_("Could not perform immediate configuration on '%s'. "
                "Please see man 5 apt.conf under APT::Immediate-Configure for details. (%d)"),Pkg.FullName().c_str(),2);
+#endif
    }
    
    return true;
@@ -991,6 +993,7 @@ pkgPackageManager::OrderResult pkgPackageManager::OrderInstall()
       }
    }
 
+#if 0
    // Final run through the configure phase
    if (ConfigureAll() == false)
       return Failed;
@@ -1005,6 +1008,7 @@ pkgPackageManager::OrderResult pkgPackageManager::OrderInstall()
 	 return Failed;
       }
    }
+#endif
 	 
    return Completed;
 }
