diff --git a/Makefile b/Makefile
index 4e0dd5b..d6912c1 100644
--- a/Makefile
+++ b/Makefile
@@ -22,24 +22,15 @@ PACKAGE_SOURCE= README efence.3 Makefile efence.h \
 OBJECTS= efence.o page.o print.o
 
 all:	libefence.a libefence.so.0.0 tstheap eftest
-	@ echo
-	@ echo "Testing Electric Fence."
-	@ echo "After the last test, it should print that the test has PASSED."
-	./eftest
-	./tstheap 3072
-	@ echo
-	@ echo "Electric Fence confidence test PASSED." 
-	@ echo
 
 install: libefence.a libefence.3 libefence.so.0.0
-	$(INSTALL) -m 755 ef.sh $(BIN_INSTALL_DIR)/ef
 	$(INSTALL) -m 644 libefence.a $(LIB_INSTALL_DIR)
 	$(INSTALL) -m 755 libefence.so.0.0 $(LIB_INSTALL_DIR)
 	- rm -f $(LIB_INSTALL_DIR)/libefence.so.0
 	ln -s libefence.so.0.0 $(LIB_INSTALL_DIR)/libefence.so.0
 	- rm -f $(LIB_INSTALL_DIR)/libefence.so
-	ln -s libefence.so.0.0 $(LIB_INSTALL_DIR)/libefence.so
-	$(INSTALL) -m 644 efence.3 $(MAN_INSTALL_DIR)/libefence.3
+	ln -s libefence.so.0 $(LIB_INSTALL_DIR)/libefence.so
+	$(INSTALL) -m 644 libefence.3 $(MAN_INSTALL_DIR)/libefence.3
 
 clean:
 	- rm -f $(OBJECTS) tstheap.o eftest.o tstheap eftest \
