diff --git a/Makefile b/Makefile
index 4e0dd5b..c615256 100644
--- a/Makefile
+++ b/Makefile
@@ -32,14 +32,13 @@ all:	libefence.a libefence.so.0.0 tstheap eftest
 	@ echo
 
 install: libefence.a libefence.3 libefence.so.0.0
-	$(INSTALL) -m 755 ef.sh $(BIN_INSTALL_DIR)/ef
 	$(INSTALL) -m 644 libefence.a $(LIB_INSTALL_DIR)
 	$(INSTALL) -m 755 libefence.so.0.0 $(LIB_INSTALL_DIR)
 	- rm -f $(LIB_INSTALL_DIR)/libefence.so.0
 	ln -s libefence.so.0.0 $(LIB_INSTALL_DIR)/libefence.so.0
 	- rm -f $(LIB_INSTALL_DIR)/libefence.so
-	ln -s libefence.so.0.0 $(LIB_INSTALL_DIR)/libefence.so
-	$(INSTALL) -m 644 efence.3 $(MAN_INSTALL_DIR)/libefence.3
+	ln -s libefence.so.0 $(LIB_INSTALL_DIR)/libefence.so
+	$(INSTALL) -m 644 libefence.3 $(MAN_INSTALL_DIR)/libefence.3
 
 clean:
 	- rm -f $(OBJECTS) tstheap.o eftest.o tstheap eftest \
