diff --git a/debian/openssh-server.sshd.pam.in b/debian/openssh-server.sshd.pam.in
index 2cad67d..fd59af8 100644
--- a/debian/openssh-server.sshd.pam.in
+++ b/debian/openssh-server.sshd.pam.in
@@ -16,7 +16,7 @@ account    required     pam_nologin.so
 # SELinux needs to be the first session rule.  This ensures that any
 # lingering context has been cleared.  Without this it is possible that a
 # module could execute code in the wrong domain.
-session [success=ok ignore=ignore module_unknown=ignore default=bad]        pam_selinux.so close
+#session [success=ok ignore=ignore module_unknown=ignore default=bad]        pam_selinux.so close
 
 # Set the loginuid process attribute.
 session    required     pam_loginuid.so
@@ -49,7 +49,7 @@ session    required     pam_env.so user_readenv=1 envfile=/etc/default/locale
 # SELinux needs to intervene at login time to ensure that the process starts
 # in the proper default security context.  Only sessions which are intended
 # to run in the user's context should be run after this.
-session [success=ok ignore=ignore module_unknown=ignore default=bad]        pam_selinux.so open
+#session [success=ok ignore=ignore module_unknown=ignore default=bad]        pam_selinux.so open
 
 # Standard Un*x password updating.
 @include common-password
