diff --git a/Makefile b/Makefile
index 22f756e..2c9be88 100644
--- a/Makefile
+++ b/Makefile
@@ -61,7 +61,7 @@ DIST_DIR := $(shell mktemp -d /tmp/glew.XXXXXX)/$(DIST_NAME)
 
 AR      ?= ar
 INSTALL ?= install
-STRIP   ?= strip
+STRIP    =
 RM      ?= rm -f
 LN      ?= ln -sf
 
@@ -267,7 +267,7 @@ endif
 
 install.bin: glew.bin
 	$(INSTALL) -d -m 0755 $(BINDIR)
-	$(INSTALL) -s -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
+	$(INSTALL) -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
 
 install.include:
 	$(INSTALL) -d -m 0755 $(INCDIR)
