From ef9451c8eda10021a591006ae75409c5e2ef7cea Mon Sep 17 00:00:00 2001
From: Toshiko YOSHIDA <toshiko.yoshida@toshiba.co.jp>
Date: Tue, 25 Apr 2017 16:28:08 +0900
Subject: [PATCH] add option to disable pacemaker

Signed-off-by: Toshiko YOSHIDA <toshiko.yoshida@toshiba.co.jp>
---
 configure.in | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/configure.in b/configure.in
index 8b6a0d6..85a97b9 100644
--- a/configure.in
+++ b/configure.in
@@ -225,10 +225,12 @@ LIBS="$ocfs_tools_save_LIBS"
 AC_SUBST(BUILD_DEBUGOCFS2)
 
 pcmk_found=
-AC_CHECK_LIB(crmcluster, crm_get_peer,
-  [AC_CHECK_HEADER(pacemaker/crm_config.h, pcmk_found=yes,
-    [AC_MSG_WARN([Pacemaker headers not found, pacemaker support will not be built])])],
-  [AC_MSG_WARN([libcrmcluster not found, pacemaker support will not be built])])
+AC_ARG_ENABLE(pacemaker, [  --enable-pacemaker=[yes/no]    Build pacemaker support [default=no]],
+  [AC_CHECK_LIB(crmcluster, crm_get_peer,
+    [AC_CHECK_HEADER(pacemaker/crm_config.h, pcmk_found=yes,
+      [AC_MSG_WARN([Pacemaker headers not found, pacemaker support will not be built])])],
+    [AC_MSG_WARN([libcrmcluster not found, pacemaker support will not be built])])],
+  [pcmk_found=no])
 
 # We use cman_replyto_shutdown to insure a new enough libcman
 cman_found=
-- 
2.1.4

