diff --git a/configure.in b/configure.in
index 1e6c503..781c713 100644
--- a/configure.in
+++ b/configure.in
@@ -372,9 +372,19 @@ if test "$FreeType2_IncRoot" = "" >/dev/null 2>&1 ; then
  ],AC_DEFINE(_NO_FREETYPE))])])
 fi
 
-AC_CHECK_FILE(/usr/include/libxml2/libxml,[CFLAGS="$CFLAGS -I/usr/include/libxml2/"],[
- AC_CHECK_FILE(/usr/local/include/libxml2/libxml,[CFLAGS="$CFLAGS -I/usr/local/include -I/usr/local/include/libxml2/"],
-  AC_DEFINE(_NO_LIBXML))])
+Libxml2_IncRoot=
+if sh -c "xml2-config --prefix" >/dev/null 2>&1 ; then
+ Libxml2_IncRoot=`xml2-config --prefix`
+ AC_CHECK_FILE($Libxml2_IncRoot/include/libxml2,CFLAGS="$CFLAGS -I$Libxml2_IncRoot/include/libxml2/",Libxml2_IncRoot=)
+ ft_flags=`xml2-config --cflags`
+ CFLAGS="$CFLAGS $ft_flags"
+fi
+
+if test "$Libxml2_IncRoot" = "" >/dev/null 2>&1 ; then
+ AC_CHECK_FILE(/usr/include/libxml2/libxml,[CFLAGS="$CFLAGS -I/usr/include/libxml2/"],[
+  AC_CHECK_FILE(/usr/local/include/libxml2/libxml,[CFLAGS="$CFLAGS -I/usr/local/include -I/usr/local/include/libxml2/"],
+   AC_DEFINE(_NO_LIBXML))])
+fi
 
 dnl Some old versions of lib(un)gif don't work for us, they've got the
 dnl Function argument in the wrong place. So check the header files.
