diff --git a/configure.in b/configure.in
index 1f6f43c..e433eb0 100644
--- a/configure.in
+++ b/configure.in
@@ -372,9 +372,8 @@ if test "$FreeType2_IncRoot" = "" >/dev/null 2>&1 ; then
  ],AC_DEFINE(_NO_FREETYPE))])])
 fi
 
-AC_CHECK_FILE(/usr/include/libxml2/libxml,[CFLAGS="$CFLAGS -I/usr/include/libxml2/"],[
- AC_CHECK_FILE(/usr/local/include/libxml2/libxml,[CFLAGS="$CFLAGS -I/usr/local/include -I/usr/local/include/libxml2/"],
-  AC_DEFINE(_NO_LIBXML))])
+ft_flags=`pkg-config --cflags libxml-2.0`
+CFLAGS="$CFLAGS $ft_flags"
 
 dnl Some old versions of lib(un)gif don't work for us, they've got the
 dnl Function argument in the wrong place. So check the header files.
