diff --git a/fontforge/python.c b/fontforge/python.c
index 26b3076..10c440a 100644
--- a/fontforge/python.c
+++ b/fontforge/python.c
@@ -9629,7 +9629,7 @@ static PyObject *fontiter_iternextkey(fontiterobject *di) {
 #ifdef _HAS_LONGLONG
 	    const char *dictfmt = "{sKsKsK}";
 #else
-	    const char *dictfmt = "{sksksk}"
+	    const char *dictfmt = "{sksksk}";
 #endif
 	    PyObject *glyph, *tempdict;
 	    PyObject *matched;
