diff --git a/bindings/perl/Makefile.am b/bindings/perl/Makefile.am
index 71b3a6f..82c2b6c 100644
--- a/bindings/perl/Makefile.am
+++ b/bindings/perl/Makefile.am
@@ -11,8 +11,8 @@ all-am: hdate.so
 	
 
 hdate.so: hdate_wrap.cxx
-	g++ -fpic -c hdate_wrap.cxx $(CFLAGS) $(CPPFLAGS) $(PE_CFLAGS) -I$(top_srcdir)/src
-	g++ -shared hdate_wrap.o ../../src/.libs/libhdate.so -o hdate.so $(LDFLAGS)
+	$(CXX) -fpic -c hdate_wrap.cxx $(CFLAGS) $(CPPFLAGS) $(PE_CFLAGS) -I$(top_srcdir)/src
+	$(CXX) -shared hdate_wrap.o ../../src/.libs/libhdate.so -o hdate.so $(LDFLAGS)
 
 hdate_wrap.cxx: hdate.i
 	swig -perl -c++ hdate.i
diff --git a/bindings/python/Makefile.am b/bindings/python/Makefile.am
index 69f8a3e..5e29346 100644
--- a/bindings/python/Makefile.am
+++ b/bindings/python/Makefile.am
@@ -11,8 +11,8 @@ all-am: _hdate.so
 	
 
 _hdate.so: hdate_wrap.cxx
-	g++ -fpic -c hdate_wrap.cxx $(CPPFLAGS) $(CFLAGS) $(PY_CFLAGS) -I$(top_srcdir)/src
-	g++ -shared hdate_wrap.o ../../src/.libs/libhdate.so -o _hdate.so $(LDFLAGS)
+	$(CXX) -fpic -c hdate_wrap.cxx $(CPPFLAGS) $(CFLAGS) $(PY_CFLAGS) -I$(top_srcdir)/src
+	$(CXX) -shared hdate_wrap.o ../../src/.libs/libhdate.so -o _hdate.so $(LDFLAGS)
 
 hdate_wrap.cxx: hdate.i
 	swig -python -c++ hdate.i
diff --git a/configure.in b/configure.in
index a2ad22f..000437f 100644
--- a/configure.in
+++ b/configure.in
@@ -97,15 +97,14 @@ AC_ARG_WITH(python-sitelib-dir,
   [path to python site lib])])
 
 if test "$have_python" = "yes"; then
-	  PY_PREFIX=`python -c 'import sys ; print sys.prefix'`
 	  PY_VERSION=`python -c 'import sys ; print sys.version[[0:3]]'`
-	  AC_MSG_CHECKING(for $PY_PREFIX/include/python$PY_VERSION/Python.h)
-	  if test -f $PY_PREFIX/include/python$PY_VERSION/Python.h; then
+	  AC_MSG_CHECKING(for ${PY_PREFIX}/include/python$PY_VERSION/Python.h)
+	  if test -f ${PY_PREFIX}/include/python$PY_VERSION/Python.h; then
 		  AC_MSG_RESULT(yes)
-		  PY_CFLAGS="-I$PY_PREFIX/include/python$PY_VERSION"
+		  PY_CFLAGS="-I${PY_PREFIX}/include/python$PY_VERSION"
 		  
 		  if test "$with_python_sitelib_dir" = "" ; then
-		    PY_MODULES_PATH="$PY_PREFIX/lib/python$PY_VERSION/dist-packages"
+		    PY_MODULES_PATH="${PY_PREFIX}/lib/python$PY_VERSION/dist-packages"
 		  else
 		    PY_MODULES_PATH=$with_python_sitelib_dir
 		  fi
diff --git a/configure.in b/configure.in
index c2813f1..dbfefca 100644
--- a/configure.in
+++ b/configure.in
@@ -226,12 +226,10 @@ AC_ARG_WITH(perl-sitelib-dir,
   [path to perl site lib])])
   
 if test "$have_perl" = "yes"; then
-	  PE_PREFIX=`(perl -e 'use Config; print $Config{archlib};')`
-	  AC_MSG_CHECKING(for $PE_PREFIX/CORE/perl.h)
-	  if test -f $PE_PREFIX/CORE/perl.h; then
+	  AC_MSG_CHECKING(for ${PE_PREFIX}/CORE/perl.h)
+	  if test -f ${PE_PREFIX}/CORE/perl.h; then
 		  AC_MSG_RESULT(yes)
-		  PE_CFLAGS="-I$PE_PREFIX/CORE "
-		  PE_CFLAGS=$PE_CFLAGS`(perl -e 'use Config; print $Config{cppflags};')`
+		  PE_CFLAGS="-I${PE_PREFIX}/CORE "
 		  
 		  if test "$with_perl_sitelib_dir" = "" ; then
 		    PE_MODULES_PATH=`(perl -e 'use Config; print $Config{sitelib};')`
