#
# We don't build manpage to avoid dependency to xmlto
#

diff --git a/Makefile.am b/Makefile.am
index b0106a4..f12c8d1 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,10 +3,10 @@ DISTCLEANFILES = src/stamp-h[0-9]* src/config.h
 bin_PROGRAMS = src/interdiff src/filterdiff src/rediff
 bin_SCRIPTS = fixcvsdiff splitdiff editdiff recountdiff unwrapdiff dehtmldiff \
 	espdiff
-man_MANS = doc/interdiff.1 doc/filterdiff.1 doc/fixcvsdiff.1 doc/rediff.1 \
-	doc/editdiff.1 doc/combinediff.1 doc/lsdiff.1 doc/splitdiff.1 \
-	doc/grepdiff.1 doc/recountdiff.1 doc/unwrapdiff.1 doc/dehtmldiff.1 \
-	doc/flipdiff.1 doc/espdiff.6
+#man_MANS = doc/interdiff.1 doc/filterdiff.1 doc/fixcvsdiff.1 doc/rediff.1 \
+#	doc/editdiff.1 doc/combinediff.1 doc/lsdiff.1 doc/splitdiff.1 \
+#	doc/grepdiff.1 doc/recountdiff.1 doc/unwrapdiff.1 doc/dehtmldiff.1 \
+#	doc/flipdiff.1 doc/espdiff.6
 
 AM_CFLAGS = -I$(srcdir)/src
 src_interdiff_SOURCES = src/interdiff.c src/util.c src/util.h src/diff.c \
@@ -21,8 +21,8 @@ src_filterdiff_LDADD = @LIBOBJS@
 src_rediff_LDADD = @LIBOBJS@
 
 # The man pages are generated from DocBook XML.
-$(man_MANS): doc/patchutils.xml
-	xmlto man -o doc $<
+#$(man_MANS): doc/patchutils.xml
+#	xmlto man -o doc $<
 
 # Special rules for combinediff, flipdiff, lsdiff and grepdiff, which are
 # just symlinks.
@@ -51,7 +51,7 @@ uninstall-hook:
 	rm -f $(DESTDIR)$(bindir)/"`echo grepdiff|sed '$(transform)'`"
 
 CLEANFILES=src/combinediff src/flipdiff src/lsdiff src/grepdiff
-MAINTAINERCLEANFILES=$(man_MANS)
+#MAINTAINERCLEANFILES=$(man_MANS)
 
 tests/combine1/run-test: src/combinediff$(EXEEXT) src/flipdiff$(EXEEXT) \
 		src/lsdiff$(EXEEXT) src/grepdiff$(EXEEXT)
@@ -209,7 +209,7 @@ XFAIL_TESTS = \
 distclean-local:
 	-rm -rf $(top_builddir)/test-arena
 
-EXTRA_DIST = $(man_MANS) \
+EXTRA_DIST = \
 	tests/common.sh tests/soak-test \
 	$(TESTS) $(XFAIL_TESTS) \
 	BUGS COPYING TODO ChangeLog \
