diff --git a/src/util/Makefile.in b/src/util/Makefile.in
index 9584bf7..7c0104c 100644
--- a/src/util/Makefile.in
+++ b/src/util/Makefile.in
@@ -137,13 +137,13 @@ Makefile: Makefile.in
 test:	$(TESTPROG)
 
 $(PCRESO): dict_pcre.o
-	gcc -shared -Wl,-soname,dict_pcre.so -o $@ $? -lpcre -L. -lutil
+	$(CC) -shared -Wl,-soname,dict_pcre.so -o $@ $? -lpcre -L. -lutil
 
 $(TCPSO): dict_tcp.o
-	gcc -shared -Wl,-soname,dict_tcp.so -o $@ $? -L. -lutil
+	$(CC) -shared -Wl,-soname,dict_tcp.so -o $@ $? -L. -lutil
 
 $(LIB):	$(OBJS)
-	gcc -shared -Wl,-soname,libpostfix-util.so.1 -o $(LIB) $(OBJS) -ldl $(SYSLIBS)
+	$(CC) -shared -Wl,-soname,libpostfix-util.so.1 -o $(LIB) $(OBJS) -ldl $(SYSLIBS)
 
 $(LIB_DIR)/$(LIB): $(LIB)
 	cp $(LIB) $(LIB_DIR)
diff --git a/src/global/Makefile.in b/src/global/Makefile.in
index c09069e..60b7372 100644
--- a/src/global/Makefile.in
+++ b/src/global/Makefile.in
@@ -131,25 +131,25 @@ Makefile: Makefile.in
 test:	$(TESTPROG)
 
 $(LIB):	$(OBJS)
-	gcc -shared -Wl,-soname,libpostfix-global.so.1 -o $(LIB) $(OBJS) $(LIBS) $(SYSLIBS)
+	$(CC) -shared -Wl,-soname,libpostfix-global.so.1 -o $(LIB) $(OBJS) $(LIBS) $(SYSLIBS)
 
 $(CDBSO): dict_cdb.o mkmap_cdb.o
-	gcc -shared -Wl,-soname,dict_cdb.so -o $@ $? -lcdb -L. -lutil
+	$(CC) -shared -Wl,-soname,dict_cdb.so -o $@ $? $(SYSLIBS) -lcdb -L. -lutil
 
 dict_cdb.o: ../util/dict_cdb.c
 	$(CC) -fPIC $(CFLAGS) -c $?
 
 $(LDAPSO): dict_ldap.o
-	gcc -shared -Wl,-soname,dict_ldap.so -o $@ $? -lldap -llber -L../../lib -lutil -L. -lglobal
+	$(CC) -shared -Wl,-soname,dict_ldap.so -o $@ $?  $(SYSLIBS) -lldap -llber -L../../lib -lutil -L. -lglobal
 
 $(MYSQLSO): dict_mysql.o
-	gcc -shared -Wl,-soname,dict_mysql.so -o $@ $? -lmysqlclient -L. -lutil -lglobal
+	$(CC) -shared -Wl,-soname,dict_mysql.so -o $@ $? $(SYSLIBS) -lmysqlclient -L. -lutil -lglobal
 
 $(PGSQLSO): dict_pgsql.o
-	gcc -shared -Wl,-soname,dict_pgsql.so -o $@ $? -lpq -L. -lutil -lglobal
+	$(CC) -shared -Wl,-soname,dict_pgsql.so -o $@ $?  $(SYSLIBS) -lpq -L. -lutil -lglobal
 
 $(SQLITESO): dict_sqlite.o
-	gcc -shared -Wl,-soname,dict_sqlite.so -o $@ $? -lsqlite3 -L. -lutil -lglobal -lpthread
+	$(CC) -shared -Wl,-soname,dict_sqlite.so -o $@ $?  $(SYSLIBS) -lsqlite3 -L. -lutil -lglobal -lpthread
 
 $(LIB_DIR)/$(LIB): $(LIB)
 	cp $(LIB) $(LIB_DIR)
diff --git a/src/master/Makefile.in b/src/master/Makefile.in
index cf44d22..f539587 100644
--- a/src/master/Makefile.in
+++ b/src/master/Makefile.in
@@ -41,7 +41,7 @@ tests:
 root_tests:
 
 $(LIB):	$(LIB_OBJ)
-	gcc -shared -Wl,-soname,libpostfix-master.so.1 -o $(LIB) $(LIB_OBJ) $(LIBS) $(SYSLIBS)
+	$(CC) -shared -Wl,-soname,libpostfix-master.so.1 -o $(LIB) $(LIB_OBJ) $(LIBS) $(SYSLIBS)
 
 $(LIB_DIR)/$(LIB): $(LIB)
 	cp $(LIB) $(LIB_DIR)/$(LIB)
diff --git a/src/dns/Makefile.in b/src/dns/Makefile.in
index f70cb96..e14c6fe 100644
--- a/src/dns/Makefile.in
+++ b/src/dns/Makefile.in
@@ -31,7 +31,7 @@ tests:	test dns_rr_to_pa_test dns_rr_to_sa_test dns_sa_to_rr_test \
 root_tests:
 
 $(LIB):	$(OBJS)
-	gcc -shared -Wl,-soname,libpostfix-dns.so.1 -o $(LIB) $(OBJS) $(LIBS) $(SYSLIBS)
+	$(CC) -shared -Wl,-soname,libpostfix-dns.so.1 -o $(LIB) $(OBJS) $(LIBS) $(SYSLIBS)
 
 $(LIB_DIR)/$(LIB): $(LIB)
 	cp $(LIB) $(LIB_DIR)
diff --git a/src/tls/Makefile.in b/src/tls/Makefile.in
index 0daa4a3..5a9a814 100644
--- a/src/tls/Makefile.in
+++ b/src/tls/Makefile.in
@@ -40,7 +40,7 @@ tests:
 root_tests:
 
 $(LIB):	$(OBJS)
-	gcc -shared -Wl,-soname,libpostfix-tls.so.1 -o $(LIB) $(OBJS) $(LIBS) $(SYSLIBS)
+	$(CC) -shared -Wl,-soname,libpostfix-tls.so.1 -o $(LIB) $(OBJS) $(LIBS) $(SYSLIBS)
 
 $(LIB_DIR)/$(LIB): $(LIB)
 	cp $(LIB) $(LIB_DIR)
