diff --git a/conf/postfix-files b/conf/postfix-files
index a43ed1f..1754207 100644
--- a/conf/postfix-files
+++ b/conf/postfix-files
@@ -65,13 +65,13 @@ $queue_directory/saved:d:$mail_owner:-:700:ucr
 $queue_directory/trace:d:$mail_owner:-:700:ucr
 $daemon_directory/anvil:f:root:-:755
 $daemon_directory/bounce:f:root:-:755
-$daemon_directory/dict_cdb.so:f:root:-:755
-$daemon_directory/dict_ldap.so:f:root:-:755
-$daemon_directory/dict_pcre.so:f:root:-:755
-$daemon_directory/dict_mysql.so:f:root:-:755
-$daemon_directory/dict_sqlite.so:f:root:-:755
-$daemon_directory/dict_tcp.so:f:root:-:755
-$daemon_directory/dict_sdbm.so:f:root:-:755
+#$daemon_directory/dict_cdb.so:f:root:-:755
+#$daemon_directory/dict_ldap.so:f:root:-:755
+#$daemon_directory/dict_pcre.so:f:root:-:755
+#$daemon_directory/dict_mysql.so:f:root:-:755
+#$daemon_directory/dict_sqlite.so:f:root:-:755
+#$daemon_directory/dict_tcp.so:f:root:-:755
+#$daemon_directory/dict_sdbm.so:f:root:-:755
 $daemon_directory/cleanup:f:root:-:755
 $daemon_directory/discard:f:root:-:755
 $daemon_directory/dnsblog:f:root:-:755
@@ -132,7 +132,7 @@ $config_directory/access:f:root:-:644:p1
 $config_directory/aliases:f:root:-:644:p1
 $config_directory/bounce.cf.default:f:root:-:644:1
 $config_directory/canonical:f:root:-:644:p1
-$config_directory/dynamicmaps.cf:f:root:-:644:p1
+#$config_directory/dynamicmaps.cf:f:root:-:644:p1
 $config_directory/cidr_table:f:root:-:644:o
 $config_directory/generic:f:root:-:644:p1
 $config_directory/generics:f:root:-:644:o
