diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 71d5fc9..0d975a2 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -61,9 +61,9 @@ endif (${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
 add_subdirectory(comparerenderer)
 add_subdirectory(endian)
 add_subdirectory(bittwiddling)
-if (NOT GRAPHITE2_NFILEFACE)
-    add_subdirectory(examples)
-endif (NOT GRAPHITE2_NFILEFACE)
+#if (NOT GRAPHITE2_NFILEFACE)
+#    add_subdirectory(examples)
+#endif (NOT GRAPHITE2_NFILEFACE)
 add_subdirectory(featuremap)
 add_subdirectory(grlist)
 add_subdirectory(json)
