diff --git a/make/makedemo.unx b/make/makedemo.unx
index ec7e4cf..9983543 100644
--- a/make/makedemo.unx
+++ b/make/makedemo.unx
@@ -227,7 +227,7 @@ cdscan : cdscan.c
 # findspl
 
 findspl : findspl.c
-	$(CC) -o findspl $(LDFLAGS) findspl.c $(LIB5) $(LIB2) $(LIB1)
+	$(CC) -o findspl $(LDFLAGS) findspl.c $(LIB5) $(LIB2) $(LIB1) $(LIB3) -lnetcli
 
 # errhdr
 
@@ -250,7 +250,7 @@ entrcmd : entrcmd.c
 
 ncbisort : sortcmd.c
 	$(CC) -o ncbisort $(LDFLAGS) sortcmd.c $(LIB23) $(LIBCOMPADJ) \
-		$(LIB1) $(OTHERLIBS)
+		$(LIB1) $(OTHERLIBS) $(LIB2)
 
 # testval
 
@@ -260,68 +260,68 @@ testval : testval.c
 # fastacmd
 
 fastacmd : fastacmd.c
-	$(CC) -o fastacmd $(LDFLAGS) fastacmd.c $(LIB23) $(LIB1)
+	$(CC) -o fastacmd $(LDFLAGS) fastacmd.c $(LIB23) $(LIB1) $(LIBCOMPADJ) $(LIB2)
 
 # formatdb
 
 formatdb : formatdb.c
-	$(CC) -o formatdb $(LDFLAGS) formatdb.c $(LIB23) $(LIB2) $(LIB1)
+	$(CC) -o formatdb $(LDFLAGS) formatdb.c $(LIB23) $(LIB2) $(LIB1) $(LIBCOMPADJ)
 
 # formatrpsdb
 
 formatrpsdb : formatrpsdb.c
 	$(CC) -o formatrpsdb $(LDFLAGS) formatrpsdb.c $(LIB23) \
-		$(LIB60) $(LIB2) $(LIB1)
+		$(LIB60) $(LIB2) $(LIB1) $(LIBCOMPADJ)
 
 # blast (reincarnation of blastall from algo/blast sources)
 
 blast : blast_driver.c
 	$(CC) -o blast $(LDFLAGS) blast_driver.c $(THREAD_OBJ) $(LIB61) \
-		$(LIB60) $(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS)
+		$(LIB60) $(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS) $(LIBCOMPADJ)
 
 debruijn : debruijn.c
-	$(CC) -o debruijn $(LDFLAGS) debruijn.c $(LIB60) $(LIB1)
+	$(CC) -o debruijn $(LDFLAGS) debruijn.c $(LIB60) $(LIB1) $(LIBCOMPADJ)
 # blastall
 
 blastall : blastall.c $(THREAD_OBJ)
 	$(CC) -o blastall $(LDFLAGS) blastall.c $(THREAD_OBJ) $(LIB61) \
-		$(LIB60) $(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS)
+		$(LIB60) $(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS) $(LIBCOMPADJ)
 
 # blastall_old  (version of blastall that does not link in new algo/blast libs).
 
 blastall_old : blastall.c $(THREAD_OBJ)
 	$(CC) -o blastall_old $(LDFLAGS) -DBLASTALL_TOOLS_ONLY blastall.c \
-		$(THREAD_OBJ) $(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS)
+		$(THREAD_OBJ) $(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS) $(LIBCOMPADJ)
 
 # blastpgp
 
 blastpgp : blastpgp.c $(THREAD_OBJ)
 	$(CC) -o blastpgp $(LDFLAGS) blastpgp.c $(THREAD_OBJ) $(LIB23) \
-		$(LIB2) $(LIB1) $(THREAD_OTHERLIBS)
+		$(LIB2) $(LIB1) $(THREAD_OTHERLIBS) $(LIBCOMPADJ)
 
 # seedtop
 
 seedtop : seedtop.c $(THREAD_OBJ)
 	$(CC) -o seedtop $(LDFLAGS) seedtop.c $(THREAD_OBJ) $(LIB23) \
-		$(LIB2) $(LIB1) $(THREAD_OTHERLIBS)
+		$(LIB2) $(LIB1) $(THREAD_OTHERLIBS) $(LIBCOMPADJ)
 
 # makemat
 
 makemat : makemat.c $(THREAD_OBJ)
 	$(CC) -o makemat $(LDFLAGS) makemat.c $(THREAD_OBJ) $(LIB23) \
-		$(LIB2) $(LIB1) $(THREAD_OTHERLIBS)
+		$(LIB2) $(LIB1) $(THREAD_OTHERLIBS) $(LIBCOMPADJ)
 
 # copymat
 
 copymat : copymat.c $(THREAD_OBJ)
 	$(CC) -o copymat $(LDFLAGS) copymat.c $(THREAD_OBJ) $(LIB60) $(LIB23) \
-		$(LIB1) $(THREAD_OTHERLIBS)
+		$(LIB1) $(THREAD_OTHERLIBS) $(LIBCOMPADJ) $(LIB2)
 
 # impala
 
 impala : profiles.c $(THREAD_OBJ)
 	$(CC) -o impala $(LDFLAGS) profiles.c $(THREAD_OBJ) $(LIB23) \
-		$(LIB2) $(LIB1) $(OTHERLIBS) $(THREAD_OTHERLIBS)
+		$(LIB2) $(LIB1) $(OTHERLIBS) $(THREAD_OTHERLIBS) $(LIBCOMPADJ)
 
 # testgen
 
@@ -342,18 +342,18 @@ packfilt : packfilt.c
 
 megablast : megablast.c $(THREAD_OBJ)
 	$(CC) -o megablast $(LDFLAGS) megablast.c $(THREAD_OBJ) $(LIB61) \
-		$(LIB60) $(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS)
+		$(LIB60) $(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS) $(LIBCOMPADJ)
 
 # vecscreen
 
 vecscreen : vecscreen.c $(THREAD_OBJ)
 	$(CC) -o vecscreen $(LDFLAGS) vecscreen.c $(THREAD_OBJ) $(LIB23) \
-		$(LIB2) $(LIB1) $(THREAD_OTHERLIBS)
+		$(LIB2) $(LIB1) $(THREAD_OTHERLIBS) $(LIBCOMPADJ)
 
 # gil2bin
 
 gil2bin : gil2bin.c 
-	$(CC) -o gil2bin $(LDFLAGS) gil2bin.c $(LIB23) $(LIB1)
+	$(CC) -o gil2bin $(LDFLAGS) gil2bin.c $(LIB23) $(LIB1) $(LIB2) $(LIBCOMPADJ)
 
 # asn2idx
 
@@ -374,13 +374,13 @@ subfuse : subfuse.c
 
 blastclust : blastclust.c $(THREAD_OBJ)
 	$(CC) -o blastclust $(LDFLAGS) blastclust.c $(THREAD_OBJ) $(LIB23) \
-		$(LIB2) $(LIB1) $(THREAD_OTHERLIBS)
+		$(LIB2) $(LIB1) $(THREAD_OTHERLIBS) $(LIBCOMPADJ)
 
 # rpsblast
 
 rpsblast : rpsblast.c $(THREAD_OBJ)
 	$(CC) -o rpsblast $(LDFLAGS) rpsblast.c $(THREAD_OBJ) $(LIB61) \
-		$(LIB60) $(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS)
+		$(LIB60) $(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS) $(LIBCOMPADJ)
 
 
 # asn2xml
diff --git a/make/makenet.unx b/make/makenet.unx
index 540891a..97a851c 100644
--- a/make/makenet.unx
+++ b/make/makenet.unx
@@ -937,14 +937,17 @@ acclinkos.o : acclinkos.c
 Cn3D : $(OBJCN3D) $(BENTREZLIBS) netentcf $(BLIB36)
 	$(CC) -o Cn3D $(LDFLAGS) $(OBJCN3D) $(LIB31) $(LIB3000) $(LIB20) \
 		$(LIB45) $(LIB22) $(LIB8) $(LIB7) \
+		$(LIB3) $(LIB23) $(LIB41) $(LIB61) $(LIB6) $(LIB60) $(LIBCOMPADJ) \
 		$(LIB400) $(LIB2) $(LIB1) $(VIBLIBS) $(OGLLIBS)
 
 ddv    : $(OBJDDV)
 	$(CC) -o ddv $(LDFLAGS) $(OBJDDV) $(LIB31) $(LIB20) $(LIB45) \
+	$(LIB3) $(LIB6) $(LIB22) $(LIB23) $(LIB41) $(LIB60) $(LIBCOMPADJ) $(LIB61) \
 	$(LIB8) $(LIB7) $(LIB4) $(LIB2) $(LIB1) $(VIBLIBS)
 
 udv    : $(OBJUDV)
 	$(CC) -o udv $(LDFLAGS) $(OBJUDV) $(LIB31) $(LIB20) \
+	$(LIB3) $(LIB6) $(LIB22) $(LIB23) $(LIB41) $(LIB60) $(LIBCOMPADJ) $(LIB61) \
 	$(LIB8) $(LIB7) $(LIB4) $(LIB2) $(LIB1) $(VIBLIBS)
 
 Nentrez : entrez.c $(ULIB31)
@@ -991,7 +994,7 @@ aceread_tst :	aceread_tst.c
 
 # asn2gb program (asn2gb)
 asn2gb :	asn2gb.c
-	$(CC) -g -o asn2gb $(LDFLAGS) asn2gb.c $(LIB23) $(LIB2) $(LIB1)
+	$(CC) -g -o asn2gb $(LDFLAGS) asn2gb.c $(LIB23) $(LIB2) $(LIB1) $(LIBCOMPADJ)
 
 # asn2gb_psf, uses PUBSEQBioseqFetchEnable instead of PubSeqFetchEnable
 # should be used only internally within NCBI.
@@ -1003,7 +1006,7 @@ asn2gb_psf :	asn2gb.c
 # asn2fsa program (asn2fsa)
 asn2fsa :	asn2fsa.c
 	$(CC) -g -o asn2fsa $(LDFLAGS) asn2fsa.c $(THREAD_OBJ) \
-		$(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS)
+		$(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS) $(LIBCOMPADJ)
 
 # asn2fsa_psf, uses PUBSEQBioseqFetchEnable instead of PubSeqFetchEnable
 # should be used only internally within NCBI.
@@ -1019,7 +1022,7 @@ asn2all :	asn2all.c
 
 # tbl2asn
 tbl2asn : tbl2asn.c 
-	$(CC) -o tbl2asn $(LDFLAGS) tbl2asn.c $(LIB23) $(LIB2) $(LIB1)
+	$(CC) -o tbl2asn $(LDFLAGS) tbl2asn.c $(LIB23) $(LIB2) $(LIB1) $(LIBCOMPADJ)
 
 # tbl2asn_psf, uses PUBSEQBioseqFetchEnable instead of PubSeqFetchEnable
 # should be used only internally within NCBI.
@@ -1059,7 +1062,7 @@ asnval_dbx_psf :	asnval.c
 # asndisc program (asndisc)
 asndisc :	asndisc.c
 	$(CC) -g -o asndisc $(LDFLAGS) asndisc.c $(THREAD_OBJ) $(LIB41) \
-		$(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS)
+		$(LIB23) $(LIB2) $(LIB1) $(THREAD_OTHERLIBS) $(LIB6) $(LIBCOMPADJ)
 
 # asndisc_psf, uses PUBSEQBioseqFetchEnable instead of PubSeqFetchEnable
 # should be used only internally within NCBI.
@@ -1083,7 +1086,7 @@ demo_aceread_tst :	aceread_tst.c
 # asnmacro
 asnmacro : asnmacro.c 
 	$(CC) -o asnmacro $(LDFLAGS) asnmacro.c $(LIB61) $(LIB60) $(LIB23) \
-		$(LIB2) $(LIB1)
+		$(LIB2) $(LIB1) $(LIBCOMPADJ)
 
 # asnstrip program (asnstrip)
 asnstrip : asnstrip.c 
@@ -1115,7 +1118,7 @@ insdseqget :	insdseqget.c
 
 # nps2gps program (nps2gps)
 nps2gps :	nps2gps.c
-	$(CC) -g -o nps2gps $(LDFLAGS) nps2gps.c $(LIB23) $(LIB2) $(LIB1)
+	$(CC) -g -o nps2gps $(LDFLAGS) nps2gps.c $(LIB23) $(LIB2) $(LIB1) $(LIBCOMPADJ)
 
 # trna2sap program (trna2sap)
 trna2sap :	trna2sap.c
@@ -1133,13 +1136,13 @@ testent2 :	testent2.c
 # network Entrez2 application (entrez2)
 entrez2 :	entrez2.c
 	$(CC) -g -o entrez2 $(LDFLAGS) entrez2.c $(LIB20) \
-		$(LIB2) $(LIB4) $(LIB1) $(VIBLIBS)
+		$(LIB2) $(LIB4) $(LIB1) $(VIBLIBS) $(LIB22) $(LIB23) $(LIB61) $(LIB41) $(LIB60) $(LIB6) $(LIBCOMPADJ)
 	$(VIB_POST_LINK) entrez2
 
 # demo program (spidey)
 spidey :	spideymain.c
 	$(CC) -o spidey $(LDFLAGS) spideymain.c $(LIB41) $(LIB23) \
-		 $(LIB2) $(LIB1)
+		 $(LIB2) $(LIB1) $(LIB6) $(LIBCOMPADJ)
 
 # demo program dotmatrix
 dotmatrix :	dotmain.c
@@ -1172,7 +1175,7 @@ asn2asn : asn2asn.c
 # demo program (cleanasn)
 
 cleanasn : cleanasn.c
-	$(CC) -o cleanasn $(LDFLAGS) cleanasn.c $(LIB23) $(LIB2) $(LIB1)
+	$(CC) -o cleanasn $(LDFLAGS) cleanasn.c $(LIB23) $(LIB2) $(LIB1) $(LIBCOMPADJ)
 
 # cleanasn_psf, uses PUBSEQBioseqFetchEnable instead of PubSeqFetchEnable
 # should be used only internally within NCBI.
@@ -1211,24 +1214,25 @@ id0test: id0test.c $(BNETCLILIB) $(BLIB13)
 	$(CC) -g -o id0test $(LDFLAGS) id0test.c $(LIB13) $(NETCLILIB) $(LIB2) $(LIB1) $(OTHERLIBS)
 
 idfetch: idfetch.c $(BNETCLILIB) $(BLIB41)
-	$(CC) -g -o idfetch $(LDFLAGS) idfetch.c $(LIB41) $(LIB2) $(LIB1)
+	$(CC) -g -o idfetch $(LDFLAGS) idfetch.c $(LIB41) $(LIB2) $(LIB1) $(LIB6)
 
 # gapped BLAST client
 # blastcl3 now has the same source code as blastall but should be compiled
 # with -DBLAST_CS_API flag
 blastcl3: blastall.c $(BNETCLILIB) $(BLIB36)
 	$(CC) -g -o blastcl3 $(LDFLAGS) -DBLAST_CS_API blastall.c \
-		$(LIB61) $(LIB60) $(LIB36) $(LIB23) $(LIB2) $(LIB1)
+		$(LIB61) $(LIB60) $(LIB36) $(LIB23) $(LIB2) $(LIB1) $(LIB6) $(LIBCOMPADJ)
 
 # BLAST 2 sequences
 # Uses network libraries to access Entrez for retrieving sequences by gi/accession
 bl2seq : bl2seq.c
 	$(CC) -o bl2seq $(LDFLAGS) bl2seq.c $(LIB61) $(LIB60) $(LIB23) \
-		$(LIB41) $(LIB2) $(LIB1)
+		$(LIB41) $(LIB2) $(LIB1) $(LIB6) $(LIBCOMPADJ)
 
 taxblast: taxblast_main.c $(BLIB41) $(BLIB40)
 	$(CC) -g -o taxblast $(LDFLAGS) taxblast_main.c \
 		$(LIB61) $(LIB60) $(LIB36) $(LIB41) $(LIB40) $(LIB23) \
+		$(LIB6) $(LIBCOMPADJ) \
 		$(LIB2) $(LIB1) $(OTHERLIBS)
 
 # test client for the suggest network service
@@ -1411,6 +1415,7 @@ Psequin : $(SRCSEQUIN) $(BENTREZLIBS) $(BLIB61) $(BLIB60) $(BLIB23) \
 	$(BLIB2) $(BLIB1) $(ULIB31)
 	$(CC) -o Psequin -I. $(LDFLAGS) -UINTERNAL_NCBI_SEQUIN $(SRCSEQUIN) \
 	$(LIB31) $(LIB20) $(LIB61) $(LIB36) $(LIB23) $(LIB11) \
+	$(LIB3) $(LIB5) $(LIB6) $(LIB22) $(LIB60) $(LIB41) $(LIBCOMPADJ) \
 	$(LIB4) $(LIB2) $(LIB1) $(VIBLIBS)
 	$(VIB_POST_LINK) Psequin
 
@@ -1421,6 +1426,7 @@ sbtedit : $(SRCSBTEDIT) $(BENTREZLIBS) $(BLIB61) $(BLIB60) $(BLIB23) \
 	$(BLIB11) $(BLIB8) $(BLIB7) $(NETCLIBLIB) $(BLIB5) $(BLIB4) $(BLIB3) \
 	$(BLIB2) $(BLIB1) $(ULIB31)
 	$(CC) -o sbtedit -I. $(LDFLAGS) -UINTERNAL_NCBI_SEQUIN $(SRCSBTEDIT) \
+	$(LIB6) $(LIB22) $(LIB41) $(LIB60) $(LIB61) $(LIBCOMPADJ) \
 	$(LIB20) $(LIB23) $(LIB4) $(LIB2) $(LIB1) $(VIBLIBS)
 	$(VIB_POST_LINK) sbtedit
 
