diff --git a/Makefile b/Makefile
index 6682f10..27869bb 100644
--- a/Makefile
+++ b/Makefile
@@ -9,31 +9,32 @@ VERS_ESONAME =1
 VERS_ESO     =$(VERS_ESONAME).0.4
 
 DESTDIR =
-prefix=/usr
-datadir=$(prefix)/share
-libdir=$(prefix)/lib
-libexecdir=$(prefix)/libexec
-bindir=$(prefix)/bin
-includedir=$(prefix)/include
-SHRDIR=$(datadir)/libustr-dev
-DOCSHRDIR=$(datadir)/doc/libustr-doc
-EXAMDIR=$(DOCSHRDIR)/examples
-mandir=$(datadir)/man
-MBINDIR=$(libexecdir)/ustr-$(VERS_FULL)
+prefix		?=/usr
+datadir		?=$(prefix)/share
+libdir		?=$(prefix)/lib
+libexecdir	?=$(prefix)/libexec
+bindir		?=$(prefix)/bin
+includedir	?=$(prefix)/include
+SHRDIR		?=$(datadir)/libustr-dev
+DOCSHRDIR	?=$(datadir)/doc/libustr-doc
+EXAMDIR		?=$(DOCSHRDIR)/examples
+mandir		?=$(datadir)/man
+MBINDIR		?=$(libexecdir)/ustr-$(VERS_FULL)
 
 ###############################################################################
 #  This is here to work around the "Fedora build system requirement" that a big
 # pile of crap ends up in the build.log file even though all sane people don't
 # want that. See rhbz#248231
 ###############################################################################
-HIDE=@
+#HIDE=@
 
-CC = cc
-AR = ar
-RANLIB = ranlib
-LDCONFIG = /sbin/ldconfig
+vsnp		?= 0
+CC		?= cc
+AR		?= ar
+RANLIB		?= ranlib
+LDCONFIG	?= /sbin/ldconfig
 
-CFLAGS  = -O2 -g
+CFLAGS		?= -O2 -g
 
 # Debug versions...
 WARNS = -W -Wall -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-align -Wwrite-strings -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wnested-externs -Wno-format-zero-length -Wformat-nonliteral -Wformat-security # -Wfloat-equal -- no floats
@@ -415,16 +416,14 @@ install: all-shared ustr.pc ustr-debug.pc
 ustr-import-multilib: ustr-import-multilib.in
 		sed -e 's,@INCLUDEDIR@,$(includedir),g' -e 's,@MBINDIR@,$(MBINDIR),g' < $< > $@
 
-install-multilib-linux: install autoconf_64b ustr-import-multilib
+install-multilib-linux: install ustr-import-multilib
 		install -d $(DESTDIR)$(MBINDIR)
-		$(HIDE)mlib=`./autoconf_64b`; \
-                   if test "x$$mlib" = "x1"; then mlib=64; else mlib=32; fi; \
-                   mv -f $(DESTDIR)$(includedir)/ustr-conf-debug.h \
-                         $(DESTDIR)$(includedir)/ustr-conf-debug-$$mlib.h; \
+		mv -f $(DESTDIR)$(includedir)/ustr-conf-debug.h \
+                         $(DESTDIR)$(includedir)/ustr-conf-debug-$(mlib).h; \
                    mv -f $(DESTDIR)$(includedir)/ustr-conf.h \
-                         $(DESTDIR)$(includedir)/ustr-conf-$$mlib.h; \
+                         $(DESTDIR)$(includedir)/ustr-conf-$(mlib).h; \
                    mv -f $(DESTDIR)$(bindir)/ustr-import \
-                         $(DESTDIR)$(MBINDIR)/ustr-import-$$mlib
+                         $(DESTDIR)$(MBINDIR)/ustr-import-$(mlib)
 		install -m 644 -t $(DESTDIR)$(includedir) $(SRC_HDRS_MULTI_LIB)
 		$(HIDE)mv -f $(DESTDIR)$(includedir)/ustr-conf-debug-multilib-linux.h $(DESTDIR)$(includedir)/ustr-conf-debug.h
 		$(HIDE)mv -f $(DESTDIR)$(includedir)/ustr-conf-multilib-linux.h $(DESTDIR)$(includedir)/ustr-conf.h
@@ -447,27 +446,16 @@ clean:
 
 distclean: clean
 		rm -f ustr-import
-		rm -f autoconf_64b autoconf_vsnprintf
 		rm -f ustr-conf.h ustr-conf-debug.h
 		rm -rf lcov-output
 
 
-ustr-import: ustr-import.in autoconf_64b autoconf_vsnprintf
+ustr-import: ustr-import.in
 		$(HIDE)echo Creating $@
-		$(HIDE)sz64=`./autoconf_64b`; vsnp=`./autoconf_vsnprintf`; \
-		sed -e 's,@INCLUDEDIR@,$(includedir),g' -e 's,@SHRDIR@,$(SHRDIR),g' -e 's,@VERS@,$(VERS),g'  -e 's,@VERS_FULL@,$(VERS_FULL),g' -e "s,@HAVE_64bit_SIZE_MAX@,$$sz64,g" -e "s,@HAVE_RETARDED_VSNPRINTF@,$$vsnp,g" < $< > $@
+		$(HIDE)if test "$(mlib)" = "64" ; then sz64=1; else sz64=0; fi; \
+		sed -e 's,@INCLUDEDIR@,$(includedir),g' -e 's,@SHRDIR@,$(SHRDIR),g' -e 's,@VERS@,$(VERS),g'  -e 's,@VERS_FULL@,$(VERS_FULL),g' -e "s,@HAVE_64bit_SIZE_MAX@,$$sz64,g" -e "s,@HAVE_RETARDED_VSNPRINTF@,$(vsnp),g" < $< > $@
 		$(HIDE)chmod 755 $@
 
-# Use CFLAGS so that CFLAGS="... -m32" does the right thing
-autoconf_64b: autoconf_64b.c
-		$(HIDE)echo Compiling: auto configuration test:  64bit
-		$(HIDE)$(CC) $(CFLAGS) -o $@ $<
-
-autoconf_vsnprintf: autoconf_vsnprintf.c
-		$(HIDE)echo Compiling: auto configuration test:  vsnprintf
-		$(HIDE)$(CC) -o $@ $<
-
-# Use LDFLAGS for LDFLAGS="-m32"
 $(OPT_LIB_SHARED): $(LIB_SHARED_OPT)
 		$(HIDE)echo Linking SO OPT lib: $@
 		$(HIDE)$(CC) -shared $^ -Wl,-soname -Wl,$(OPT_LIB_SHARED_NAME) -Wl,-version-script -Wl,libustr.ver $(LDFLAGS) -o $@
@@ -485,23 +473,21 @@ libustr-debug.a: $(LIB_STATIC_DBG)
 		$(HIDE)$(AR) ru $@ $^
 		$(HIDE)$(RANLIB) $@
 
-ustr-conf.h: ustr-conf.h.in autoconf_64b autoconf_vsnprintf
+ustr-conf.h: ustr-conf.h.in
 		$(HIDE)echo Creating $@
 		$(HIDE)have_stdint_h=0; dbg1=0; dbg2=0; \
-                sz64=`./autoconf_64b`; vsnp=`./autoconf_vsnprintf`; \
-                if test -f "/usr/include/stdint.h"; then have_stdint_h=1; fi; \
+                $(HIDE)if test "$(mlib)" = "64" ; then sz64=1; else sz64=0; fi; \
                 if test -f "$(prefix)/include/stdint.h"; then have_stdint_h=1; fi; \
                 if test -f "$(includedir)/stdint.h"; then have_stdint_h=1; fi; \
-		sed -e "s,@HAVE_STDINT_H@,$$have_stdint_h,g" -e "s,@USE_ASSERT@,$$dbg1,g" -e "s,@USE_EOS_MARK@,$$dbg2,g" -e "s,@HAVE_64bit_SIZE_MAX@,$$sz64,g" -e "s,@HAVE_RETARDED_VSNPRINTF@,$$vsnp,g" < $< > $@
+		sed -e "s,@HAVE_STDINT_H@,$$have_stdint_h,g" -e "s,@USE_ASSERT@,$$dbg1,g" -e "s,@USE_EOS_MARK@,$$dbg2,g" -e "s,@HAVE_64bit_SIZE_MAX@,$$sz64,g" -e "s,@HAVE_RETARDED_VSNPRINTF@,$(vsnp),g" < $< > $@
 
-ustr-conf-debug.h: ustr-conf.h.in autoconf_64b autoconf_vsnprintf
+ustr-conf-debug.h: ustr-conf.h.in
 		$(HIDE)echo Creating $@
 		$(HIDE)have_stdint_h=0; dbg1=1; dbg2=1; \
-                sz64=`./autoconf_64b`; vsnp=`./autoconf_vsnprintf`; \
-                if test -f "/usr/include/stdint.h"; then have_stdint_h=1; fi; \
+		$(HIDE)if test "$(mlib)" = "64" ; then sz64=1; else sz64=0; fi; \
                 if test -f "$(prefix)/include/stdint.h"; then have_stdint_h=1; fi; \
                 if test -f "$(includedir)/stdint.h"; then have_stdint_h=1; fi; \
-		sed -e "s,@HAVE_STDINT_H@,$$have_stdint_h,g" -e "s,@USE_ASSERT@,$$dbg1,g" -e "s,@USE_EOS_MARK@,$$dbg2,g" -e "s,@HAVE_64bit_SIZE_MAX@,$$sz64,g" -e "s,@HAVE_RETARDED_VSNPRINTF@,$$vsnp,g" < $< > $@
+		sed -e "s,@HAVE_STDINT_H@,$$have_stdint_h,g" -e "s,@USE_ASSERT@,$$dbg1,g" -e "s,@USE_EOS_MARK@,$$dbg2,g" -e "s,@HAVE_64bit_SIZE_MAX@,$$sz64,g" -e "s,@HAVE_RETARDED_VSNPRINTF@,$(vsnp),g" < $< > $@
 
 
 #  We don't use $(DEPS_NONC_ALL) as deps. because it then recompiles everything
