diff --git a/Makefile.am b/Makefile.am
index 8f9e014..6adbfa6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -37,8 +37,7 @@ SUBDIRS =					\
 	test-suite				\
 	benchmark-suite				\
 	gc-benchmarks				\
-	am					\
-	doc
+	am
 
 libguileincludedir = $(pkgincludedir)/$(GUILE_EFFECTIVE_VERSION)
 libguileinclude_HEADERS = libguile.h
diff --git a/configure.ac b/configure.ac
index 3969929..0b12cbb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1661,9 +1661,6 @@ AC_CONFIG_FILES([
   lib/Makefile
   benchmark-suite/Makefile
   gc-benchmarks/Makefile
-  doc/Makefile
-  doc/r5rs/Makefile
-  doc/ref/Makefile
   emacs/Makefile
   examples/Makefile
   libguile/Makefile
