diff --git a/lib/Makefile.am b/lib/Makefile.am
index bcdd0d7..0628d97 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -46,54 +46,35 @@ BUILT_SOURCES = actiontabs.h errtabs.h fieldtabs.h flagtabs.h \
 	ftypetabs.h i386_tables.h ia64_tables.h machinetabs.h \
 	msg_typetabs.h optabs.h ppc_tables.h s390_tables.h \
 	s390x_tables.h x86_64_tables.h
-if USE_ALPHA
 BUILT_SOURCES += alpha_tables.h
-endif
-if USE_ARM
 BUILT_SOURCES += arm_tables.h
-endif
-if USE_AARCH64
 BUILT_SOURCES += aarch64_tables.h
-endif
 noinst_PROGRAMS = gen_actiontabs_h gen_errtabs_h gen_fieldtabs_h \
 	gen_flagtabs_h gen_ftypetabs_h gen_i386_tables_h \
 	gen_ia64_tables_h gen_machinetabs_h gen_msg_typetabs_h \
 	gen_optabs_h gen_ppc_tables_h gen_s390_tables_h \
 	gen_s390x_tables_h gen_x86_64_tables_h
-if USE_ALPHA
 noinst_PROGRAMS += gen_alpha_tables_h
-endif
-if USE_ARM
 noinst_PROGRAMS += gen_arm_tables_h
-endif
-if USE_AARCH64
 noinst_PROGRAMS += gen_aarch64_tables_h
-endif
 gen_actiontabs_h_SOURCES = gen_tables.c gen_tables.h actiontab.h
 gen_actiontabs_h_CFLAGS = $(AM_CFLAGS) '-DTABLE_H="actiontab.h"'
 actiontabs.h: gen_actiontabs_h Makefile
 	./gen_actiontabs_h --lowercase --i2s --s2i action > $@
 
-if USE_ALPHA
 gen_alpha_tables_h_SOURCES = gen_tables.c gen_tables.h alpha_table.h
 gen_alpha_tables_h_CFLAGS = $(AM_CFLAGS) '-DTABLE_H="alpha_table.h"'
 alpha_tables.h: gen_alpha_tables_h Makefile
 	./gen_alpha_tables_h --lowercase --i2s --s2i alpha_syscall > $@
-endif
 
-if USE_ARM
 gen_arm_tables_h_SOURCES = gen_tables.c gen_tables.h arm_table.h
 gen_arm_tables_h_CFLAGS = $(AM_CFLAGS) '-DTABLE_H="arm_table.h"'
 arm_tables.h: gen_arm_tables_h Makefile
 	./gen_arm_tables_h --lowercase --i2s --s2i arm_syscall > $@
-endif
-
-if USE_AARCH64
 gen_aarch64_tables_h_SOURCES = gen_tables.c gen_tables.h aarch64_table.h
 gen_aarch64_tables_h_CFLAGS = $(AM_CFLAGS) '-DTABLE_H="aarch64_table.h"'
 aarch64_tables.h: gen_aarch64_tables_h Makefile
 	./gen_aarch64_tables_h --lowercase --i2s --s2i aarch64_syscall > $@
-endif
 
 gen_errtabs_h_SOURCES = gen_tables.c gen_tables.h errtab.h
 gen_errtabs_h_CFLAGS = $(AM_CFLAGS) '-DTABLE_H="errtab.h"'
