diff --git a/swig/auditswig.i b/swig/auditswig.i
index acc332b..4918083 100644
--- a/swig/auditswig.i
+++ b/swig/auditswig.i
@@ -37,8 +37,8 @@ signed
 #define __attribute(X) /*nothing*/
 typedef unsigned __u32;
 typedef unsigned uid_t;
-%include "/usr/include/linux/audit.h"
+%include "(STAGING_INCDIR)/linux/audit.h"
 #define __extension__ /*nothing*/
-%include "/usr/include/stdint.h"
+%include "(STAGING_INCDIR)/stdint.h"
 %include "../lib/libaudit.h"
 
