diff --git a/swig/Makefile.in b/swig/Makefile.in
index e99cfc0..94b41ca 100644
--- a/swig/Makefile.in
+++ b/swig/Makefile.in
@@ -359,7 +359,7 @@ top_srcdir = @top_srcdir@
 EXTRA_DIST = auditswig.i
 CONFIG_CLEAN_FILES = *.loT *.rej *.orig
 AM_CFLAGS = -fPIC -DPIC -fno-strict-aliasing
-INCLUDES = -I. -I$(top_builddir) -I${top_srcdir}/lib $(PYTHON_INCLUDES)
+INCLUDES = -I. -I$(top_builddir) -I${top_srcdir}/lib $(PYTHON_INCLUDES) -I(STAGING_INCDIR)/python2.7 -I(STAGING_INCDIR_audit)/linux
 pyexec_PYTHON = audit.py
 pyexec_LTLIBRARIES = _audit.la
 pyexec_SOLIBRARIES = _audit.so
