diff --git a/Makefile b/Makefile
index d583b51..010dba1 100644
--- a/Makefile
+++ b/Makefile
@@ -35,9 +35,6 @@ statduplcheck.o: statduplcheck.c statduplcheck.h
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c statduplcheck.c
 
 install: whichman ftff ftwhich $(MANP)
-	$(STRIP) whichman
-	$(STRIP) ftwhich
-	$(STRIP) ftff
 	[ -d "$(prefix)/bin" ] || $(INSTALL) -d $(prefix)/bin
 	[ -d "$(mandir)/man1" ] || $(INSTALL) -d $(mandir)/man1
 	$(INSTALL) -m 755 whichman $(prefix)/bin
