diff --git a/elf/ldconfig.c b/elf/ldconfig.c
index b66bcf6..8680c1a 100644
--- a/elf/ldconfig.c
+++ b/elf/ldconfig.c
@@ -322,6 +322,7 @@ parse_opt (int key, char *arg, struct argp_state *state)
   return 0;
 }
 
+#define REPORT_BUGS_TO "mailing list : poky@yoctoproject.org"
 /* Print bug-reporting information in the help message.  */
 static char *
 more_help (int key, const char *text, void *input)
@@ -346,7 +347,7 @@ For bug reporting instructions, please see:\n\
 static void
 print_version (FILE *stream, struct argp_state *state)
 {
-  fprintf (stream, "ldconfig %s%s\n", PKGVERSION, VERSION);
+  fprintf (stream, "ldconfig (Hacked Poky Version)\n");
   fprintf (stream, gettext ("\
 Copyright (C) %s Free Software Foundation, Inc.\n\
 This is free software; see the source for copying conditions.  There is NO\n\
