diff --git a/Makefile b/Makefile
index 3a432c8..c833a14 100644
--- a/Makefile
+++ b/Makefile
@@ -79,7 +79,7 @@ DEFS = -DDEBIAN -DUSE_SIGCHLD $(DEBUG_DEFS) $(PAM_DEFS) $(SELINUX_DEFS) $(AUDIT_
 #DEFS		=	-D_BSD_SIGNALS -Dconst=
 #<<the name of the BSD-like install program>>
 #INSTALL = installbsd
-INSTALL = install -s
+INSTALL = install
 #<<any special load flags>>
 # LDFLAGS		=	-s
 # Let install do the strip
