diff --git a/contrib/slapd-modules/acl/Makefile b/contrib/slapd-modules/acl/Makefile
index a381981..980121c 100644
--- a/contrib/slapd-modules/acl/Makefile
+++ b/contrib/slapd-modules/acl/Makefile
@@ -6,8 +6,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = 
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/addpartial/Makefile b/contrib/slapd-modules/addpartial/Makefile
index fc6c7c9..fdcf8a6 100644
--- a/contrib/slapd-modules/addpartial/Makefile
+++ b/contrib/slapd-modules/addpartial/Makefile
@@ -6,8 +6,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = 
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/allop/Makefile b/contrib/slapd-modules/allop/Makefile
index 7812707..702d283 100644
--- a/contrib/slapd-modules/allop/Makefile
+++ b/contrib/slapd-modules/allop/Makefile
@@ -6,8 +6,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = 
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/allowed/Makefile b/contrib/slapd-modules/allowed/Makefile
index c961e26..bb3677d 100644
--- a/contrib/slapd-modules/allowed/Makefile
+++ b/contrib/slapd-modules/allowed/Makefile
@@ -18,8 +18,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = -DSLAPD_OVER_ALLOWED=SLAPD_MOD_DYNAMIC
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/autogroup/Makefile b/contrib/slapd-modules/autogroup/Makefile
index b49aaa0..7b76e1a 100644
--- a/contrib/slapd-modules/autogroup/Makefile
+++ b/contrib/slapd-modules/autogroup/Makefile
@@ -2,12 +2,12 @@
 
 LDAP_SRC = ../../..
 LDAP_BUILD = ../../..
-LDAP_INC = -I$(LDAP_BUILD)/debian/build/include -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/slapd
-LDAP_LIB = $(LDAP_BUILD)/debian/build/libraries/libldap_r/libldap_r.la \
-	$(LDAP_BUILD)/debian/build/libraries/liblber/liblber.la
+LDAP_INC = -I##STAGING_INCDIR## -I$(LDAP_BUILD)/debian/build/include -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/slapd
+LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
+	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/debian/build/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = 
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/cloak/Makefile b/contrib/slapd-modules/cloak/Makefile
index 4f116bc..d6e830e 100644
--- a/contrib/slapd-modules/cloak/Makefile
+++ b/contrib/slapd-modules/cloak/Makefile
@@ -6,8 +6,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = -DSLAPD_OVER_CLOAK=SLAPD_MOD_DYNAMIC
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/comp_match/Makefile b/contrib/slapd-modules/comp_match/Makefile
index f1c6424..fda2e5e 100644
--- a/contrib/slapd-modules/comp_match/Makefile
+++ b/contrib/slapd-modules/comp_match/Makefile
@@ -29,8 +29,8 @@ SSL_DIR = /usr/local
 SSL_INC = -I$(SSL_DIR)/include/openssl
 SSL_LIB = -lcrypto -L$(SSL_DIR)/lib
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = -DLDAP_COMPONENT
 INCS = $(LDAP_INC) $(SNACC_INC) $(SSL_INC)
diff --git a/contrib/slapd-modules/denyop/Makefile b/contrib/slapd-modules/denyop/Makefile
index 5a2864c..cae61b9 100644
--- a/contrib/slapd-modules/denyop/Makefile
+++ b/contrib/slapd-modules/denyop/Makefile
@@ -6,8 +6,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = -DSLAPD_OVER_DENYOP=SLAPD_MOD_DYNAMIC
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/dsaschema/Makefile b/contrib/slapd-modules/dsaschema/Makefile
index 4cc4953..272c8e6 100644
--- a/contrib/slapd-modules/dsaschema/Makefile
+++ b/contrib/slapd-modules/dsaschema/Makefile
@@ -6,8 +6,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = 
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/dupent/Makefile b/contrib/slapd-modules/dupent/Makefile
index 14937d4..a18d41c 100644
--- a/contrib/slapd-modules/dupent/Makefile
+++ b/contrib/slapd-modules/dupent/Makefile
@@ -18,8 +18,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = -DSLAPD_OVER_DUPENT=SLAPD_MOD_DYNAMIC
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/kinit/Makefile b/contrib/slapd-modules/kinit/Makefile
index 549a39f..c7eb56b 100644
--- a/contrib/slapd-modules/kinit/Makefile
+++ b/contrib/slapd-modules/kinit/Makefile
@@ -6,8 +6,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = 
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/lastbind/Makefile b/contrib/slapd-modules/lastbind/Makefile
index 3b26461..e263ce4 100644
--- a/contrib/slapd-modules/lastbind/Makefile
+++ b/contrib/slapd-modules/lastbind/Makefile
@@ -12,12 +12,12 @@
 
 LDAP_SRC = ../../..
 LDAP_BUILD = ../../..
-LDAP_INC = -I$(LDAP_BUILD)/debian/build/include -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/slapd
-LDAP_LIB = $(LDAP_BUILD)/debian/build/libraries/libldap_r/libldap_r.la \
-	$(LDAP_BUILD)/debian/build/libraries/liblber/liblber.la
+LDAP_INC = -I##STAGING_INCDIR## -I$(LDAP_BUILD)/debian/build/include -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/slapd
+LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
+	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/debian/build/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = -DSLAPD_OVER_LASTBIND=SLAPD_MOD_DYNAMIC
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/lastmod/Makefile b/contrib/slapd-modules/lastmod/Makefile
index 521011a..293e96c 100644
--- a/contrib/slapd-modules/lastmod/Makefile
+++ b/contrib/slapd-modules/lastmod/Makefile
@@ -6,8 +6,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = -DSLAPD_OVER_LASTMOD=SLAPD_MOD_DYNAMIC
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/noopsrch/Makefile b/contrib/slapd-modules/noopsrch/Makefile
index d59c6e6..df2ce3a 100644
--- a/contrib/slapd-modules/noopsrch/Makefile
+++ b/contrib/slapd-modules/noopsrch/Makefile
@@ -18,8 +18,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = -DSLAPD_OVER_NOOPSRCH=SLAPD_MOD_DYNAMIC
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/nops/Makefile b/contrib/slapd-modules/nops/Makefile
index 5cf0375..10bbdf2 100644
--- a/contrib/slapd-modules/nops/Makefile
+++ b/contrib/slapd-modules/nops/Makefile
@@ -6,8 +6,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = -DSLAPD_OVER_NOPS=SLAPD_MOD_DYNAMIC
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/nssov/Makefile b/contrib/slapd-modules/nssov/Makefile
index 28c8003..8d0231a 100644
--- a/contrib/slapd-modules/nssov/Makefile
+++ b/contrib/slapd-modules/nssov/Makefile
@@ -25,9 +25,9 @@ LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 
 NLDAPD_INC=-Inss-pam-ldapd
 
-LIBTOOL = $(LDAP_BUILD)/libtool
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
 OPT = -g -O2 -Wall
-CC = gcc
+CC = ##TARGET_SYS##-gcc
 DEFS =
 INCS = $(LDAP_INC) $(NLDAPD_INC)
 LIBS = $(LDAP_LIB)
diff --git a/contrib/slapd-modules/passwd/Makefile b/contrib/slapd-modules/passwd/Makefile
index e3def2b..9ddd6a9 100644
--- a/contrib/slapd-modules/passwd/Makefile
+++ b/contrib/slapd-modules/passwd/Makefile
@@ -6,8 +6,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = 
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/passwd/pbkdf2/Makefile b/contrib/slapd-modules/passwd/pbkdf2/Makefile
index 64ad97c..70f0441 100644
--- a/contrib/slapd-modules/passwd/pbkdf2/Makefile
+++ b/contrib/slapd-modules/passwd/pbkdf2/Makefile
@@ -6,8 +6,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 #DEFS = -DSLAPD_PBKDF2_DEBUG

diff --git a/contrib/slapd-modules/proxyOld/Makefile b/contrib/slapd-modules/proxyOld/Makefile
index ea252ef..1648613 100644
--- a/contrib/slapd-modules/proxyOld/Makefile
+++ b/contrib/slapd-modules/proxyOld/Makefile
@@ -18,8 +18,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = 
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/samba4/Makefile b/contrib/slapd-modules/samba4/Makefile
index 529c0e9..ae4406d 100644
--- a/contrib/slapd-modules/samba4/Makefile
+++ b/contrib/slapd-modules/samba4/Makefile
@@ -18,8 +18,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = -DSLAPD_OVER_RDNVAL=SLAPD_MOD_DYNAMIC \
 	-DSLAPD_OVER_PGUID=SLAPD_MOD_DYNAMIC \
diff --git a/contrib/slapd-modules/smbk5pwd/Makefile b/contrib/slapd-modules/smbk5pwd/Makefile
index be25b2c..514276f 100644
--- a/contrib/slapd-modules/smbk5pwd/Makefile
+++ b/contrib/slapd-modules/smbk5pwd/Makefile
@@ -14,9 +14,9 @@
 
 LDAP_SRC = ../../..
 LDAP_BUILD = ../../..
-LDAP_INC = -I$(LDAP_BUILD)/debian/build/include -I$(LDAP_BUILD)/debian/build/servers/slapd -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/slapd
-LDAP_LIB = $(LDAP_BUILD)/debian/build/libraries/libldap_r/libldap_r.la \
-	$(LDAP_BUILD)/debian/build/libraries/liblber/liblber.la
+LDAP_INC = -I##STAGING_INCDIR## -I$(LDAP_BUILD)/debian/build/include -I$(LDAP_BUILD)/debian/build/servers/slapd -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/slapd
+LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
+	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
 SSL_INC = 
 SSL_LIB = -lnettle
@@ -24,8 +24,8 @@ SSL_LIB = -lnettle
 HEIMDAL_INC = $(shell krb5-config.heimdal --cflags kadm-server)
 HEIMDAL_LIB = $(shell krb5-config.heimdal --libs kadm-server)
 
-LIBTOOL = $(LDAP_BUILD)/debian/build/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 # Omit DO_KRB5, DO_SAMBA or DO_SHADOW if you don't want to support it.
 DEFS = -DDO_KRB5 -DDO_SAMBA -DDO_SHADOW
diff --git a/contrib/slapd-modules/trace/Makefile b/contrib/slapd-modules/trace/Makefile
index 8167374..bf58978 100644
--- a/contrib/slapd-modules/trace/Makefile
+++ b/contrib/slapd-modules/trace/Makefile
@@ -6,8 +6,8 @@ LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/s
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = -DSLAPD_OVER_TRACE=SLAPD_MOD_DYNAMIC
 INCS = $(LDAP_INC)
diff --git a/contrib/slapd-modules/passwd/sha2/Makefile b/contrib/slapd-modules/passwd/sha2/Makefile
index 381530a..23337f2 100644
--- a/contrib/slapd-modules/passwd/sha2/Makefile
+++ b/contrib/slapd-modules/passwd/sha2/Makefile
@@ -3,11 +3,11 @@
 LDAP_SRC = ../../../..
 LDAP_BUILD = ../../../..
 LDAP_INC = -I$(LDAP_BUILD)/debian/build/include -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/slapd
-LDAP_LIB = $(LDAP_BUILD)/debian/build/libraries/libldap_r/libldap_r.la \
-	$(LDAP_BUILD)/debian/build/libraries/liblber/liblber.la
+LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
+	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
-LIBTOOL = $(LDAP_BUILD)/debian/build/libtool
-CC = gcc
+LIBTOOL = $(LDAP_BUILD)/##TARGET_SYS##-libtool
+CC = ##TARGET_SYS##-gcc
 OPT = -g -O2 -Wall
 DEFS = 
 #DEFS = -DSLAPD_SHA2_DEBUG
