diff --git a/configure.in b/configure.in
index 74956b1..91b769d 100644
--- a/configure.in
+++ b/configure.in
@@ -134,7 +134,7 @@ AC_ARG_ENABLE(x11,
 
 if test "$enable_x11" = "yes"; then
   CFLAGS_saved="$CFLAGS"
-  CFLAGS="$CFLAGS -I/usr/X11R6/include"
+  CFLAGS="$CFLAGS"
   AC_CHECK_HEADER(X11/X.h, x11_found=yes, x11_found=no)
   CFLAGS="$CFLAGS_saved"
   if test "$x11_found" = no; then
@@ -142,8 +142,8 @@ if test "$enable_x11" = "yes"; then
     AC_MSG_WARN([
 *** no X11/X.h found -- building without X11 support.])
   else
-    X11_LIBS="-L/usr/X11R6/lib -lX11 -lXext"
-    X11_CFLAGS="-I/usr/X11R6/include"
+    X11_LIBS="-lX11 -lXext"
+    X11_CFLAGS=""
   fi
 fi
 
@@ -236,32 +236,32 @@ case "$target_or_host" in
     need_libc_r=yes
     need_libdl=no
     want_ppcasm=yes
-    CPPFLAGS="$CPPFLAGS -I/usr/local/include"
-    LDFLAGS="$LDFLAGS -L/usr/local/lib"
+    CPPFLAGS="$CPPFLAGS"
+    LDFLAGS="$LDFLAGS"
     ;;
 
   *-openbsd*)
     need_libc_r=yes
     need_libdl=no
     want_ppcasm=no
-    CPPFLAGS="$CPPFLAGS -I/usr/local/include"
-    LDFLAGS="$LDFLAGS -L/usr/local/lib"
+    CPPFLAGS="$CPPFLAGS"
+    LDFLAGS="$LDFLAGS"
     ;;
 
   *-netbsd*)
     need_libc_r=no
     need_libdl=no
     want_ppcasm=yes
-    CPPFLAGS="$CPPFLAGS -I/usr/pkg/include"
-    LDFLAGS="$LDFLAGS -L/usr/pkg/lib"
+    CPPFLAGS="$CPPFLAGS"
+    LDFLAGS="$LDFLAGS"
     ;;    
 
   *-darwin*)
     need_libc_r=no
     need_libdl=yes
     want_ppcasm=no
-    CPPFLAGS="$CPPFLAGS -I/sw/include"
-    LDFLAGS="$LDFLAGS -L/sw/lib"
+    CPPFLAGS="$CPPFLAGS"
+    LDFLAGS="$LDFLAGS"
     ;;
 
   sh-*-elf)
